<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="unawaz1996" />

<meta name="date" content="2023-09-19" />

<title>Metadata-attributes</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.31/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BITHub-preprocessing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proprocessing datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-bulk-preprocess-data.html">Bulk RNA-seq datasets</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Dataset characteristics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="03-metadata-attributes.html">Metadata attributes</a>
    </li>
    <li>
      <a href="04-bulk-deconvolution.html">Deconvolution of bulk RNA-seq datasets</a>
    </li>
    <li class="dropdown-header">Drivers of variation</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/unawaz1996/BITHub-preprocessing">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Metadata-attributes</h1>
<h4 class="author">unawaz1996</h4>
<h4 class="date">2023-09-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BITHub-preprocessing/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230919code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230919)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230919code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230919)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996BITHubpreprocessingtreea450b04a57e549a928e1aa7059291f74d8437bedtargetblanka450b04a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/tree/a450b04a57e549a928e1aa7059291f74d8437bed" target="_blank">a450b04</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996BITHubpreprocessingtreea450b04a57e549a928e1aa7059291f74d8437bedtargetblanka450b04a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/unawaz1996/BITHub-preprocessing/tree/a450b04a57e549a928e1aa7059291f74d8437bed" target="_blank">a450b04</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    output/Thesis_plots/Variance/PCA/BrainSpan/brainspan-varPart-cor/

Unstaged changes:
    Deleted:    HDBR-PC-cor-filt.pdf
    Modified:   analysis/_site.yml
    Deleted:    cerebroViz_output_outer_1.svg
    Deleted:    cerebroViz_output_slice_1.svg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/03-metadata-attributes.Rmd</code>) and HTML
(<code>docs/03-metadata-attributes.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/BITHub-preprocessing/20c235ad9071bcb20693e5cf468fba1c60976e4f/docs/03-metadata-attributes.html" target="_blank">20c235a</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/7007796461ad04b7464369609f5b9977d7947023/analysis/03-metadata-attributes.Rmd" target="_blank">7007796</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
wflow_publish(c("analysis/04-bulk-deconvolution.Rmd",
"analysis/03-metadata-attributes.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/BITHub-preprocessing/727f567b309dfb3439af0311b3def4d6b11a1d97/docs/03-metadata-attributes.html" target="_blank">727f567</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/BITHub-preprocessing/8033da061f4701908ab78412c157616fcb7819fa/docs/03-metadata-attributes.html" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/95607d00a209cfcf49c174cea4c8bd9e0be0184b/analysis/03-metadata-attributes.Rmd" target="_blank">95607d0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
wflow_publish(c("analysis/03-metadata-attributes.Rmd",
"analysis/index.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1">
<h1>Set-up</h1>
<pre class="r"><code>library(recount3)
library(magrittr)
library(tibble)
library(reshape2)
library(SummarizedExperiment)
library(corrplot)
library(dplyr)
library(ggvenn)
library(pander)
library(gridExtra)
library(variancePartition)
library(DT)
library(EnsDb.Hsapiens.v86)
library(singscore)
library(AnnotationHub)
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(DT)
library(naniar)
source(&quot;code/preprocess/functions.R&quot;)
source(&quot;code/preprocess/def_stages.R&quot;)
library(UpSetR)
library(patchwork)
library(pheatmap)
library(ggpubr)</code></pre>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>To allow comparison of gene expression with a metadata variable of
interest, BITHub contains comprehensive metadata annotations of the
curated datasets. There are three main categories of annotation are
present in BITHub:</p>
<ul>
<li><p>Phenotype annotations: Relating to the phenotype and origin of
sample, including sex, donorIDs, age of donor, diagnosis and
ethnicity</p></li>
<li><p>Sequencing metrics These annotations relate to the quality of the
RNA-seq samples and how they were experimentally prepared (if
available).</p></li>
<li><p>Sample characteristics These include the charateristics of the
samples.</p></li>
</ul>
<p>In order to ensure the metadata information is displayed in a
user-friendly manner, highly correlated metadata annotations will be
removed and a subset will be used for the site.</p>
</div>
<div id="filtering-of-redudant-metadata-labels-via-correlation-analysis"
class="section level1">
<h1>Filtering of redudant metadata labels via correlation analysis</h1>
<p>Several datasets on BITHub contain redundant metadata variables
related to sequencing metrics. These datasets include the BrainSeq,
GTEx, HDBR and the snRNA-seq datasets including HCA and Aldinger et
al. To remove these, a correlation analysis of all sequencing metrics
was performed. Correlation of metadata was prepared after processing the
raw metadata and expression files. If you are interested in that part of
the pipeline, please refer to the README.md in the Github repo.</p>
<pre class="r"><code>dir = file.path(&quot;/home/neuro/Documents/BrainData/Bulk&quot;)
sn_dir = file.path(&quot;/home/neuro/Documents/BrainData/single-cell&quot;)
md = list()

md$BrainSeq = read.csv(file.path(dir,&quot;Brainseq/Formatted/BrainSeq-metadata.csv&quot;), header=TRUE, check.names = FALSE, row.names = 1)
md$BrainSpan = read.csv(file.path(dir,&quot;BrainSpan/Formatted/BrainSpan-metadata.csv&quot;), header=TRUE, check.names = FALSE, row.names = 1)
md$GTEx =  read.csv(file.path(dir,&quot;GTEx/Formatted/GTEx-metadata.csv&quot;), header=TRUE, check.names = FALSE, row.names =1)
md$HDBR = read.csv(file.path(dir, &quot;HDBR/Formatted/HDBR-metadata.csv&quot;), header=TRUE, check.names = FALSE, row.names =1)
#md$FANTOM5 = read.csv(file.path(dir, &quot;Fantom5/Formatted/FANTOM5-metadata.csv&quot;), header=TRUE, check.names = FALSE)
md$PsychEncode = read.csv(file.path(dir, &quot;PsychEncode/Formatted/PsychEncode-metadata.csv&quot;), header=TRUE, check.names = FALSE)
md$Velmeshev = read.csv(file.path(dir, &quot;HDBR/Formatted/HDBR-metadata.csv&quot;), header=TRUE, check.names = FALSE)
md$Aldringer = read.csv(file.path(sn_dir, &quot;aldinger/Aldringer-metadata.csv&quot;), header=TRUE, check.names = FALSE)</code></pre>
<pre class="r"><code>annot = list()

annot$BrainSpan = read.csv(here::here(&quot;data/annotations/BrainSpan-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;BrainSpan&quot;))
annot$BrainSeq = read.csv(here::here(&quot;data/annotations/BrainSeq-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;BrainSeq&quot;))

annot$GTEx = read.csv(here::here(&quot;data/annotations/GTEx-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;GTEx&quot;))

annot$PsychEncode =  read.csv(here::here(&quot;data/annotations/PsychEncode-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;PsychEncode&quot;))

annot$FANTOM5 = read.csv(here::here(&quot;data/annotations/Fantom5-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;FANTOM5&quot;))

annot$HDBR = read.csv(here::here(&quot;data/annotations/HDBR-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;HDBR&quot;))

annot$Velmeshev = read.csv(here::here(&quot;data/annotations/Velmeshev-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;Velmeshev&quot;))

annot$Aldringer = read.csv(here::here(&quot;data/annotations/Aldringer-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;Aldringer&quot;))

annot$HCA =  read.csv(here::here(&quot;data/annotations/HCA-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;HCA&quot;))

annot$Cameron =  read.csv(here::here(&quot;data/annotations/Cameron-metadata-annot.csv&quot;), header=TRUE) %&gt;%
  mutate(Dataset = c(&quot;Cameron&quot;))</code></pre>
<div id="brainseq" class="section level2">
<h2>BrainSeq</h2>
<pre class="r"><code>annot_seq = annot$BrainSeq %&gt;%
  dplyr::filter(Type == &quot;Sequencing metrics&quot;)
md.clean = md$BrainSeq %&gt;% 
  dplyr::select(contains(annot_seq$BITColumnName))
md.clean = md.clean %&gt;% select_if(~ !any(is.na(.)))
md.clean = md.clean[vapply(md.clean, function(x) length(unique(x)) &gt; 1, logical(1L))]</code></pre>
<pre class="r"><code>M = cor(data.matrix(md.clean), use = &quot;complete.obs&quot;)
#pdf(here::here(&quot;output/Thesis_plots/Metadata/BrainSeq-cor-seq-metadata.pdf&quot;), height = 10, width=10)
corrplot(M, order=&#39;hclust&#39;,
         method = &quot;circle&quot;, 
         number.cex = .50, 
         tl.cex=0.75)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-2-1.png" alt="Correlation plot showing highly correlating metadata variables in BrainSeq" width="672" />
<p class="caption">
Correlation plot showing highly correlating metadata variables in
BrainSeq
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-2-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<p>The final BrainSeq sequencing metrics contain the following
columns:</p>
<pre class="r"><code>annot$BrainSeq %&gt;% 
  dplyr::filter(Include..Yes.No....Interest == &quot;Yes&quot;) %&gt;% 
  DT::datatable()</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-7d3add13c6a483e279a1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7d3add13c6a483e279a1">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28"],["X","trimmed","numReads","numUnmapped","overallMapRate","concordMapRate","mitoMapped","mitoRate","totalAssignedGene","rRNA_rate","BrNum","Region","RIN","Age","Sex","Race","Dx","Fetal_replicating","Fetal_quiescent","OPC","Neurons","Astrocytes","Oligodendrocytes","Microglia","Endothelial",null,null,null],["SampleID","trimmed","TotalNReads","numUnmapped","MappingRate","concordMapRate","mitoMapped","mito_ratio","totalAssignedGene","rRNA_ratio","DonorID","StructureAcronym","RIN","AgeNumeric","Sex","Ethnicity","Diagnosis","Dev.Replicating","Dev.Quiescent","Adult.OPC","Adult.Neurons","Adult.Astrocytes","Adult.Oligo","Adult.Microglia","Adult.Endothelial","AgeInterval","Period","Regions"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["Sample charactertics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Phenotype","Sample charactertics","Sequencing metrics","Phenotype","Phenotype","Phenotype","Phenotype","Cell-Type","Cell-Type","Cell-Type","Cell-Type","Cell-Type","Cell-Type","Cell-Type","Cell-Type","Phenotype","Phenotype","Sample charactertics"],["BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq","BrainSeq"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>OriginalMetadataColumnName<\/th>\n      <th>BITColumnName<\/th>\n      <th>Include..Yes.No....Interest<\/th>\n      <th>Type<\/th>\n      <th>Dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"OriginalMetadataColumnName","targets":1},{"name":"BITColumnName","targets":2},{"name":"Include..Yes.No....Interest","targets":3},{"name":"Type","targets":4},{"name":"Dataset","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="gtex" class="section level2">
<h2>GTEx</h2>
<pre class="r"><code>annot_seq = annot$GTEx %&gt;%
  dplyr::filter(Type == &quot;Sequencing metrics&quot;)
md.clean = md$GTEx %&gt;%
  dplyr::select(contains(annot_seq$BITColumnName))
md.clean = md.clean %&gt;% select_if(~ !any(is.na(.)))
md.clean = md.clean[vapply(md.clean, function(x) length(unique(x)) &gt; 1, logical(1L))]</code></pre>
<pre class="r"><code>#pdf(here::here(&quot;output/Thesis_plots/Metadata/GTEx-cor-seq-metadata.pdf&quot;), height = 10, width=10)
M = cor(data.matrix(md.clean), use = &quot;complete.obs&quot;)
corrplot(M, order=&#39;hclust&#39;,
         method = &quot;circle&quot;,
         number.cex = .50,
         tl.cex=0.75)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-5-1.png" alt="Correlation plot showing highly correlating metadata variables in GTEx" width="672" />
<p class="caption">
Correlation plot showing highly correlating metadata variables in GTEx
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-5-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>annot$GTEx %&gt;% 
  dplyr::filter(Include..Yes.No....Interest == &quot;Yes&quot;) %&gt;% 
  DT::datatable()</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-4741f8913ab2a8a92f14" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4741f8913ab2a8a92f14">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48"],["SAMPID","SMRIN","SMTSISCH","AGE","SEX","SMATSSCR","SMNABTCH","SMNABTCHT","SMNABTCHD","SMGEBTCH","SMGEBTCHD","SMGEBTCHT","SMCENTER","SMTS","SMTSD","SMTSPAX","SME2MPRT","SMCHMPRS","SMNTRART","SMMAPRT","SMEXNCRT","SMGNSDTC","SMESTLBS","SMMPPD","SMNTERRT","SMRRNANM","SMRDTTL","SMTRSCPT","SMMPPDPR","SMUNPDRD","SMNTRNRT","SMMPUNRT","SMEXPEFF","SMMPPDUN","SME2MMRT","SME2ANTI","SME2SNSE","SME1ANTI","SME1SNSE","SME1PCTS","SMRRNART","SME1MPRT","SMDPMPRT","SME2PCTS","SUBJID","DTHHRDY",null,null],["SampleID","RIN","PMI","AgeInterval","Sex","AutolysisScore","SequencingBatch","TypeofBatch","DateofBatch","Genotype_or_Expression_Batch_ID","DateofGenotypeorExpressionBatch","TypeofGenotypeorExpressionBatch","BSS_Collection_side_code","Regions","StructureAcronym","Time_spent_in_PAXgene_fixative","End_2_mapping_rate","ChimericPairs","IntragenicRate","MappingRate","ExonicRate","GenesDetected","LibrarySize","ReadsMapped","IntergenicRate","rRNA","TotalNReads","TranscriptsDetected","MappedPairs","UnpairedReads","IntronicRate","MappingRate_unique","ExpressionProfilingEfficiency","MappedUnique_no_dup_flags","End_2_Mismatch_Rate","End_2_Antisense","End_Sense_2","End_1_Antisense","End_1_Sense","End_1_Sense_percentage","rRNA_ratio","End_1_Mapping_rate","DuplicationRateMapped","Percentage_IntragenicEnd_2_Reads","DonorID","HardyScale","Period","Diagnosis"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["Sample charactertics","Sequencing metrics","Sequencing metrics","Phenotype","Phenotype","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sample charactertics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Phenotype","Phenotype","Phenotype","Phenotype"],["GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx","GTEx"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>OriginalMetadataColumnName<\/th>\n      <th>BITColumnName<\/th>\n      <th>Include..Yes.No....Interest<\/th>\n      <th>Type<\/th>\n      <th>Dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"OriginalMetadataColumnName","targets":1},{"name":"BITColumnName","targets":2},{"name":"Include..Yes.No....Interest","targets":3},{"name":"Type","targets":4},{"name":"Dataset","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<!-- <!-- - FANTOM5 -->
<p>–&gt;</p>
<!-- <!-- ```{r f5-cor} -->
<p>–&gt; <!-- <!-- annot_seq = annot$FANTOM5 %>% --> –&gt;
<!-- <!--   dplyr::filter(Type == "Sequencing") --> –&gt;
<!-- <!-- md.clean = md$FANTOM5 %>%  --> –&gt;
<!-- <!--   dplyr::select(contains(annot_seq$BITHubName)) --> –&gt;
<!-- <!-- md.clean = md.clean %>% select_if(~ !any(is.na(.))) --> –&gt;
<!-- <!-- md.clean = md.clean[vapply(md.clean, function(x) length(unique(x)) > 1, logical(1L))] -->
–&gt; <!-- <!-- ``` --> –&gt;</p>
<!-- <!-- ```{r} -->
<p>–&gt;
<!-- <!-- M = cor(data.matrix(md.clean), use = "complete.obs") --> –&gt;
<!-- <!-- corrplot(M, order='hclust', --> –&gt;
<!-- <!--          method = "circle",  --> –&gt;
<!-- <!--          number.cex = .50,  --> –&gt;
<!-- <!--          tl.cex=0.75, type="upper") --> –&gt;
<!-- <!-- ``` --> –&gt;</p>
</div>
<div id="hdbr" class="section level2">
<h2>HDBR</h2>
<pre class="r"><code>hdbr_annot_seq = annot$HDBR %&gt;%
  dplyr::filter(Type == &quot;Sequencing metrics&quot;)

md.seq = md$HDBR %&gt;%
  dplyr::select(contains(hdbr_annot_seq$BITColumnName)) %&gt;%
  dplyr::select(-c(&quot;Percentage_fragment_mapped_unique_exon_fc&quot;,
                   &quot;Total_fragments_assigned_exon_fc&quot;,
                   &quot;Total_fragments_count_unique_exon_fc&quot;,
                   &quot;Total_fragments_count_unique_assigned_exon_fc&quot;,
                   &quot;AUC_all_annotated_exons&quot;, &quot;AUC_all_annotated_exons_unique&quot;,
                   &quot;BigWigFile&quot;, &quot;CellLinePrediction&quot;,&quot;PredictionType&quot;,
                   &quot;CuratedTypePrediction&quot;, &quot;PredictionType&quot;,
                   &quot;Metadata_source&quot;, &quot;Metadata_source&quot;, &quot;LibraryName&quot;,
                   &quot;LibraryStrategy&quot;, &quot;LibrarySource&quot;, &quot;LibraryLayout&quot;,
                   &quot;LibrarySelection&quot;, &quot;CellTypePrediction&quot;, &quot;PatternPredictionType&quot;,
                   &quot;SampleAccPrediction&quot;, &quot;CuratedTissuePrediction&quot;,
                   &quot;Mapping_speed_per_hour_STAR&quot;, &quot;ReadInfo&quot;, &quot;RunAlias&quot;,
                   &quot;Run_Center_Name&quot;, &quot;Run_Broker_name&quot;, &quot;Run_Center&quot;, &quot;Processed_Recount&quot;,
                   &quot;FileSource_Recount&quot;, &quot;Organism_recount&quot;, &quot;PlatformModel&quot;,
                   &quot;SampleAttributes&quot;, &quot;ExperimentAttributes&quot;, &quot;SampleName&quot;,
                   &quot;SampleTitle&quot;, &quot;RunPublished&quot;, &quot;Size&quot;,
                   &quot;MeanFragmentLength_BAM&quot;, &quot;Total_fragments_input_fc_exon_fc&quot;))

md.seq.clean = md.seq  %&gt;% select_if(~ !any(is.na(.)))
md.seq.clean = md.seq.clean[vapply(md.seq.clean, function(x) length(unique(x)) &gt; 1, logical(1L))]</code></pre>
<pre class="r"><code>#M = cor(data.matrix(md.seq.clean), use = &quot;complete.obs&quot;)
#pdf(here::here(&quot;output/Thesis_plots/Metadata/HDBR-corplot.pdf&quot;), height = 25, width=25)
corrplot(M, order=&#39;hclust&#39;,
         method = &quot;circle&quot;,
         number.cex = .50,
         cl.cex = 3,
         tl.cex=1.2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-8-1.png" alt="Correlation plot showing highly correlating metadata variables in HDBR" width="2400" />
<p class="caption">
Correlation plot showing highly correlating metadata variables in HDBR
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-8-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>annot$HDBR %&gt;% 
  dplyr::filter(Include..Yes.No....Interest == &quot;Yes&quot;) %&gt;% 
  DT::datatable()</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-521e7420bfa881ce433c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-521e7420bfa881ce433c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39"],["Run","Sample.Characteristic.block.","Sample.Characteristic.Ontology.Term.block.","Sample.Characteristic.individual.","Sample.Characteristic.karyotype.","Sample.Characteristic.organism.part.","Sample.Characteristic.sampling.site.","external_id","recount_qc.aligned_reads..chrm","recount_qc.aligned_reads..chrx","recount_qc.aligned_reads..chry","recount_qc.bc_auc.all_.","recount_qc.bc_auc.unique_.","recount_qc.bc_frag.mode_length_count","recount_qc.exon_fc.all_.","recount_qc.exon_fc_count_all.assigned","recount_qc.exon_fc_count_unique.total","recount_qc.exon_fc_count_unique.assigned","recount_qc.gene_fc_count_all.assigned","recount_qc.intron_sum_.","recount_qc.star.._of_reads_mapped_to_multiple_loci","recount_qc.star.._of_reads_mapped_to_too_many_loci","recount_qc.star.._of_reads_unmapped._other","recount_qc.star.._of_reads_unmapped._too_many_mismatches","recount_qc.star.all_mapped_reads","recount_qc.star.number_of_input_reads","recount_qc.star.uniquely_mapped_reads_.","recount_qc.star.uniquely_mapped_reads_number","recount_qc.junction_count","recount_qc.star.all_mapped_reads_both","recount_qc.star.._mapped_reads_both","recount_qc.star.._reads_mapped_to_multiple_loci_both","recount_qc.star.._reads_mapped_to_too_many_loci_both",null,null,null,"time sample in transit (minutes)",null,null],["SampleID","SequencingBatch","Block","DonorID","Sex","StructureAcronym","Hemisphere","SampleID","ChimericPairs","Percent_aligned_ChrX","Percent_aligned_ChrY","AUC_all_percentage","AUC_unique_percentage","ModeFragmentLengthCount","ExonicRate","Total_fragments_assigned_exon_fc","Total_fragments_count_unique_exon_fc","Total_fragments_count_unique_assigned_exon_fc","totalAssignedGene","IntronicRate","Percentage_mapped_multi_loci_STAR","Percentage_mapped_too_many_loci_STAR","Percentage_unmapped_other_STAR","Percentage_unmapped_too_many_mismatches_STAR","ReadsMapped","TotalNReads","MappingRate","MappingRate_unique","Junction_count","MappedPairs","MappingRate_both","Percentage_mapped_multi_loci_both_STAR","Percentage_mapped_too_many_loci_both_STAR","AgeInterval","Diagnosis","Period","PMI","Regions","AgeNumeric"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes"],["Sample charactertics","Sequencing metrics","Sequencing metrics","Phenotype","Phenotype","Sample charactertics","Sample charactertics","Sample charactertics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Sequencing metrics","Phenotype","Phenotype","Phenotype","Sequencing metrics","Sample charactertics","Phenotype"],["HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR","HDBR"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>OriginalMetadataColumnName<\/th>\n      <th>BITColumnName<\/th>\n      <th>Include..Yes.No....Interest<\/th>\n      <th>Type<\/th>\n      <th>Dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"OriginalMetadataColumnName","targets":1},{"name":"BITColumnName","targets":2},{"name":"Include..Yes.No....Interest","targets":3},{"name":"Type","targets":4},{"name":"Dataset","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="single-nucleus-rna-seq-dataset" class="section level2">
<h2>Single-nucleus RNA-seq dataset</h2>
<p>The redundant metadata columns have been removed for BITHub, however,
the metadata have not been fully explored in the BITHub thesis
chapter.</p>
<div id="aldinger-et-al-2022" class="section level3">
<h3>Aldinger et al (2022)</h3>
<pre class="r"><code>aldringer_annot_seq = annot$Aldringer %&gt;%
  dplyr::filter(Type == &quot;Sequencing metrics&quot;)

md.seq = md$Aldringer %&gt;%
  dplyr::select(contains(aldringer_annot_seq$BITColumnName))

md.seq.clean = md.seq  %&gt;% select_if(~ !any(is.na(.)))
md.seq.clean = md.seq.clean[vapply(md.seq.clean, function(x) length(unique(x)) &gt; 1, logical(1L))]

M = cor(data.matrix(md.seq.clean ), use = &quot;complete.obs&quot;)
corrplot(M, order=&#39;hclust&#39;,
         method = &quot;circle&quot;,
         number.cex = .50,
         tl.cex=0.75, type=&quot;upper&quot;)</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-10-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="exploratory-analysis-of-metadata" class="section level1">
<h1>Exploratory analysis of metadata</h1>
<p>The removal of these redundant metadata attributes resulted in 91
unique metadata across all bulk RNA-seq datasets. These metadata can be
identified in `data/Tables/Supplementary_Table_2.xlsx’.</p>
<div id="overlap-of-metadata-across-bulk-rna-seq-datasets"
class="section level2">
<h2>Overlap of metadata across bulk RNA-seq datasets</h2>
<p>As metadata were categorized into a concept that they represented
(i.e Sequencing Metrics, Phenotype attributes, Sample characteristics),
I overlapped metadata belonging to each of these categories were then
overlapped to determine the set of common metadata attributes across all
datasets.</p>
<pre class="r"><code>annot = lapply(annot, function(x){
  x %&lt;&gt;%
    dplyr::filter(Include..Yes.No....Interest == &quot;Yes&quot;)
  return(x)
  })


md$BrainSeq %&lt;&gt;% dplyr::select(contains(annot$BrainSeq$BITColumnName))
md$BrainSpan %&lt;&gt;% dplyr::select(contains(annot$BrainSpan$BITColumnName))
md$GTEx %&lt;&gt;% dplyr::select(contains(annot$GTEx$BITColumnName))
md$HDBR %&lt;&gt;% dplyr::select(contains(annot$HDBR$BITColumnName))
md$PsychEncode %&lt;&gt;% dplyr::select(contains(annot$PsychEncode$BITColumnName))</code></pre>
<pre class="r"><code>seq.metrics = lapply(annot, function(x){
  x = x %&gt;%
     dplyr::filter(Type == &quot;Sequencing metrics&quot;)
  return(x)
})


pheno.metrics = lapply(annot, function(x){
  x = x %&gt;%
     dplyr::filter(Type == &quot;Phenotype&quot;)
  return(x)
})


annot = lapply(annot, function(x){
  x = x %&gt;%
    mutate(Type = gsub(&quot;Sample charactertics&quot;, &quot;Sample characteristics&quot;, Type))
  return(x)
})
sample.metrics = lapply(annot, function(x){
  x = x %&gt;%
     dplyr::filter(Type == &quot;Sample characteristics&quot;)
  return(x)
})</code></pre>
<div id="sequencing-metrics" class="section level3">
<h3>Sequencing metrics</h3>
<pre class="r"><code>library(VennDiagram)
v= list( &quot;BrainSeq&quot; = seq.metrics$BrainSeq$BITColumnName,
          &quot;BrainSpan&quot; = seq.metrics$BrainSpan$BITColumnName,
          &quot;GTEx&quot; = seq.metrics$GTEx$BITColumnName,
          &quot;HDBR&quot; = seq.metrics$HDBR$BITColumnName,
          &quot;PsychEncode&quot; = seq.metrics$PsychEncode$BITColumnName
       #  &quot;HCA&quot; = hca.tech$BITHubColumnName,
        # &quot;Cameron et al&quot; = cam.tech$BITHubColumnName,
         #&quot;Velmeshev et al&quot; = vel.tech$BITColumnName,
         #&quot;Aldringer et al&quot; = adl.tech$BITHubColumnName
         #,
        #  &quot;Ramaker&quot; = ram.tech$BITColumnName
         )


a=calculate.overlap(v)

plot =UpSetR::upset(fromList(v), sets.bar.color = c(&quot;#7DA1D4&quot;,&quot;#E11F28&quot;,&quot;#F1605F&quot;,&quot;#49215F&quot;),
              main.bar.color = c(&quot;#7DA1D4&quot;,&quot;#E11F28&quot;,&quot;#F1605F&quot;,&quot;#49215F&quot;,
                                 &quot;#808080&quot;,&quot;#808080&quot;,&quot;#808080&quot;,&quot;#808080&quot;,
                                 &quot;#808080&quot;,&quot;#808080&quot;),
              mainbar.y.label = &quot;Overlap of Sequencing Metrics&quot;,
              sets.x.label = &quot;Total Sequencing Metrics&quot;,
                text.scale = c(1.2, 2, 1.2),
              point.size = 4)</code></pre>
<pre class="r"><code>plot</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-14-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>combos &lt;- Reduce(c,lapply(2:length(v),
                          function(x) combn(1:length(v),x,simplify=FALSE) ))
intersect &lt;- lapply(combos, function(x) Reduce(intersect,v[x]) )</code></pre>
</div>
<div id="phenotype-attributes" class="section level3">
<h3>Phenotype attributes</h3>
<pre class="r"><code>v= list( &quot;BrainSeq&quot; = pheno.metrics$BrainSeq$BITColumnName,
          &quot;BrainSpan&quot; = pheno.metrics$BrainSpan$BITColumnName,
          &quot;GTEx&quot; = pheno.metrics$GTEx$BITColumnName,
          &quot;HDBR&quot; = pheno.metrics$HDBR$BITColumnName,
          &quot;PsychEncode&quot; = pheno.metrics$PsychEncode$BITColumnName
       #  &quot;HCA&quot; = hca.tech$BITHubColumnName,
        # &quot;Cameron et al&quot; = cam.tech$BITHubColumnName,
         #&quot;Velmeshev et al&quot; = vel.tech$BITColumnName,
         #&quot;Aldringer et al&quot; = adl.tech$BITHubColumnName
         #,
        #  &quot;Ramaker&quot; = ram.tech$BITColumnName
         )



UpSetR::upset(fromList(v), sets.bar.color = c(&quot;#FAAD79&quot;,&quot;#F1605F&quot;,&quot;#E11F28&quot;,&quot;#7DA1D4&quot;,&quot;#49215F&quot;),
              main.bar.color = c(&quot;#FAAD79&quot;,&quot;#7DA1D4&quot;,
                                &quot;#808080&quot;,&quot;#808080&quot;,&quot;#808080&quot;),
              mainbar.y.label = &quot;Overlap of Phenotype Attributes&quot;,
              sets.x.label = &quot;Total Phenotype Attributes&quot;,
                text.scale = c(1.2, 2, 1.2),
              point.size = 4)</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-15-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>combos &lt;- Reduce(c,lapply(2:length(v),
                          function(x) combn(1:length(v),x,simplify=FALSE) ))
intersect &lt;- lapply(combos, function(x) Reduce(intersect,v[x]) )</code></pre>
<pre class="r"><code>v= list( &quot;BrainSeq&quot; = pheno.metrics$BrainSeq$BITColumnName,
          &quot;BrainSpan&quot; = pheno.metrics$BrainSpan$BITColumnName,
          &quot;GTEx&quot; = pheno.metrics$GTEx$BITColumnName,
          &quot;HDBR&quot; = pheno.metrics$HDBR$BITColumnName,
          &quot;PsychEncode&quot; = pheno.metrics$PsychEncode$BITColumnName
       #  &quot;HCA&quot; = hca.tech$BITHubColumnName,
        # &quot;Cameron et al&quot; = cam.tech$BITHubColumnName,
         #&quot;Velmeshev et al&quot; = vel.tech$BITColumnName,
         #&quot;Aldringer et al&quot; = adl.tech$BITHubColumnName
         #,
        #  &quot;Ramaker&quot; = ram.tech$BITColumnName
         )



UpSetR::upset(fromList(v), sets.bar.color = c(&quot;#FAAD79&quot;,&quot;#F1605F&quot;,&quot;#E11F28&quot;,&quot;#7DA1D4&quot;,&quot;#49215F&quot;),
              main.bar.color = c(&quot;#FAAD79&quot;,&quot;#7DA1D4&quot;,
                                &quot;#808080&quot;,&quot;#808080&quot;,&quot;#808080&quot;),
              mainbar.y.label = &quot;Overlap of Phenotype Attributes&quot;,
              sets.x.label = &quot;Total Phenotype Attributes&quot;,
                text.scale = c(1.2, 2, 1.2),
              point.size = 4)</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-16-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>combos &lt;- Reduce(c,lapply(2:length(v),
                          function(x) combn(1:length(v),x,simplify=FALSE) ))
intersect &lt;- lapply(combos, function(x) Reduce(intersect,v[x]) )</code></pre>
</div>
<div id="sample-characteristics" class="section level3">
<h3>Sample characteristics</h3>
<pre class="r"><code>v= list( &quot;BrainSeq&quot; = sample.metrics$BrainSeq$BITColumnName,
          &quot;BrainSpan&quot; = sample.metrics$BrainSpan$BITColumnName,
          &quot;GTEx&quot; = sample.metrics$GTEx$BITColumnName,
          &quot;HDBR&quot; = sample.metrics$HDBR$BITColumnName,
          &quot;PsychEncode&quot; = sample.metrics$PsychEncode$BITColumnName
       #  &quot;HCA&quot; = hca.tech$BITHubColumnName,
        # &quot;Cameron et al&quot; = cam.tech$BITHubColumnName,
         #&quot;Velmeshev et al&quot; = vel.tech$BITColumnName,
         #&quot;Aldringer et al&quot; = adl.tech$BITHubColumnName
         #,
        #  &quot;Ramaker&quot; = ram.tech$BITColumnName
         )


UpSetR::upset(fromList(v), sets.bar.color = c(&quot;#7DA1D4&quot;,&quot;#FAAD79&quot;,&quot;#E11F28&quot;,&quot;#F1605F&quot;,&quot;#49215F&quot;),
              main.bar.color = c(&quot;#7DA1D4&quot;,&quot;#FAAD79&quot;,
             &quot;#808080&quot;,&quot;#808080&quot;),
              mainbar.y.label = &quot;Overlap of Sample Characteristics&quot;,
              sets.x.label = &quot;Total Sample Characteristics&quot;,
                text.scale = c(1.2, 2, 1.2),
              point.size = 4)</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-17-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>combos &lt;- Reduce(c,lapply(2:length(v),
                          function(x) combn(1:length(v),x,simplify=FALSE) ))
intersect &lt;- lapply(combos, function(x) Reduce(intersect,v[x]) )</code></pre>
</div>
</div>
<div
id="determining-number-of-missing-metadata-from-each-bulk-rna-seq-analysis"
class="section level2">
<h2>Determining number of missing metadata from each bulk RNA-seq
analysis</h2>
<p>I assessed the quality of the metadata by determining the extent of
missing information across the available and retained metadata columns
in all datasets.</p>
<pre class="r"><code>count_na &lt;- function(x) sum(is.na(x))
# md$ %&gt;%
#   mutate(count_na = apply(., 1, count_na)) %&gt;%
#   ggplot(aes(count_na)) + geom_density()

na_calc = lapply(md[1:5], function(x){
  x %&lt;&gt;%
    mutate(count_na = apply(., 1, count_na)) %&gt;%
    dplyr::select(count_na)
  return(x)
})


total_nb = do.call(rbind, na_calc) %&gt;%
  rownames_to_column(&quot;Dataset&quot;) %&gt;%
  dplyr::filter(count_na != 0)</code></pre>
<pre class="r"><code>#ggsave(filename = here::here(&quot;output/Thesis_plots/Metadata/missing_md.svg&quot;), height=6, width = 7, units = &quot;in&quot;)
do.call(rbind, na_calc) %&gt;%
  rownames_to_column(&quot;Dataset&quot;) %&gt;%
  mutate(Dataset = gsub(&quot;\\..*&quot;, &quot;&quot;, Dataset)) %&gt;%
   ggplot(aes(count_na, fill= Dataset)) +
  geom_density(alpha = 0.75) +
  theme_bw()  +
     scale_fill_manual(values =c(&quot;BrainSeq&quot; =&quot;#F1605F&quot;,
                               &quot;BrainSpan&quot; = &quot;#49215F&quot;,
                               &quot;GTEx&quot; = &quot;#7DA1D4&quot;,
                               &quot;HDBR&quot; = &quot;#E11F28&quot;,
                               &quot;PsychEncode&quot; = &quot;#FAAD79&quot;))+
  xlab(&quot;Number of missing metadata&quot;) +
  ylab(&quot;Density of samples&quot;) +
  theme(axis.text.x = element_text(size = 12, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.text.y = element_text(size = 15, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=15, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;)</code></pre>
<p><img src="figure/03-metadata-attributes.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/8033da061f4701908ab78412c157616fcb7819fa/docs/figure/03-metadata-attributes.Rmd/unnamed-chunk-19-1.png" target="_blank">8033da0</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<!-- # ```{r} -->
<!-- # md$GTEx %>% apply(., 2, count_na)  -->
<!-- # md$HDBR %>% apply(.,2, count_na)  -->
<!-- # md$PsychEncode%>% apply(.,2, count_na) -->
<!-- # md$BrainSpan %>% apply(.,2, count_na)  -->
<!-- #  -->
<!-- # ``` -->
<!-- ## Heatmap showing metadata variables in at least 2 datasets -->
<!-- ```{r} -->
<!-- metadata_summary = lapply(annot, function(x) { x %>% -->
<!--     dplyr::filter(`Include..Yes.No....Interest` == "Yes") %>% -->
<!--     dplyr::select("BITColumnName",`Include..Yes.No....Interest`, "Dataset")}) -->
<!-- metadata_heatmap = metadata_summary  %>% do.call(rbind,.) %>% -->
<!--   pivot_wider(id_cols = "Dataset", names_from = "BITColumnName", values_from = "Include..Yes.No....Interest") %>% -->
<!--   as.data.frame() -->
<!-- metadata_heatmap$SampleID[6] <- "Yes" -->
<!-- metadata_heatmap[metadata_heatmap == "Yes"] <- 1 -->
<!-- metadata_heatmap[is.na(metadata_heatmap)] <- 0 -->
<!--   #mutate(comparison=gsub("\\.[0-9]*$","", comparison)) %>% -->
<!--   #xtabs(fgsea.NES ~ comparison + fgsea.Geneset,.) -->
<!-- metadata_heatmap <- apply(metadata_heatmap,2,as.character) -->
<!-- #write.csv(metadata_heatmap,here::here("output/metadata_heatmap.csv")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- heatmap_file = read.csv("output/metadata_heatmap.csv", header=TRUE) -->
<!-- md.summary = heatmap_file %>% -->
<!--   column_to_rownames("Dataset") %>% -->
<!--   pheatmap(cellheight = 15, -->
<!--              cellwidth = 15, -->
<!--              cluster_rows = FALSE, -->
<!--              cluster_cols = FALSE, -->
<!--              border_color = "#FFF8F9", -->
<!--              color = c("#FFF8F9", "black")) -->
<!-- ggsave(file = here::here("output/Thesis_plots/Metadata/summary-heatmap.svg"),  md.summary, -->
<!--        height = 15, width = 19.9329, units = "cm") -->
<!-- ``` -->
<!-- ## Shared sequencing metrics -->
<!-- ```{r} -->
<!-- x= list("HCA" = hca.tech$BITHubColumnName, -->
<!--          "Cameron et al" = cam.tech$BITHubColumnName, -->
<!--          "Velmeshev et al" = vel.tech$BITColumnName, -->
<!--          "Aldringer et al" = adl.tech$BITHubColumnName -->
<!--          #, -->
<!--         #  "Ramaker" = ram.tech$BITColumnName -->
<!--          ) -->
<!-- UpSetR::upset(fromList(x), sets.bar.color = c ("#E85571","#838DE0","#FFD256","#75C7A0")) -->
<!-- ``` -->
<!-- ### -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainspan = read.csv("/home/neuro/Documents/BrainData/Bulk/BrainSpan/Formatted/BrainSpan-metadata-subset.csv", header=TRUE, check.names = FALSE, row.names =1) -->
<!-- brainspan_decon = read.csv("/home/neuro/Documents/BrainData/Bulk/BrainSpan/Formatted/BrainSpan-decon.csv", header=TRUE, check.names = FALSE, row.names =1) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainspan_full = brainspan %>% -->
<!--   left_join(brainspan_decon %>% -->
<!--               dplyr::select("SampleID", "Neurons", -->
<!--                             "Astrocytes", "Oligodendrocytes", -->
<!--                             "Microglia", "Endothelia"), by = "SampleID") -->
<!-- write.csv(brainspan_full ,"/home/neuro/Documents/BrainData/Bulk/BrainSpan/Formatted/BrainSpan-metadata-all.csv" ) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainseq = read.csv("/home/neuro/Documents/BrainData/Bulk/Brainseq/Formatted/BrainSeq-metadata-subset.csv", header=TRUE, check.names = FALSE, row.names =1) -->
<!-- brainseq_decon = read.csv("/home/neuro/Documents/BrainData/Bulk/Brainseq/Formatted/BrainSeq-decon.csv", header=TRUE, check.names = FALSE, row.names =1) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainseq_full = brainseq %>% -->
<!--   dplyr::select(-c("Dev.Replicating", "Dev.Quiescent", "Adult.OPC", "Adult.Neurons", "Adult.Astrocytes", "Adult.Oligo", "Adult.Microglia", "Adult.Endothelial")) %>% -->
<!--   left_join(brainseq_decon %>% -->
<!--               dplyr::select("SampleID", "Neurons", -->
<!--                             "Astrocytes", "Oligodendrocytes", -->
<!--                             "Microglia", "Endothelia"), by = "SampleID") -->
<!-- write.csv(brainseq_full ,"/home/neuro/Documents/BrainData/Bulk/Brainseq/Formatted/BrainSeq-metadata-all.csv" ) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- gtex = read.csv("/home/neuro/Documents/BrainData/Bulk/GTEx/Formatted/GTEx-metadata-subset.csv", header=TRUE, check.names = FALSE, row.names =1) -->
<!-- gtex_decon = read.csv("/home/neuro/Documents/BrainData/Bulk/GTEx/Formatted/GTEx-decon.csv", header=TRUE, check.names = FALSE, row.names =1) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- gtex_full = gtex %>% -->
<!--   left_join(gtex_decon %>% -->
<!--               dplyr::select("SampleID", "Neurons", -->
<!--                             "Astrocytes", "Oligodendrocytes", -->
<!--                             "Microglia", "Endothelia"), by = "SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- write.csv(gtex_full ,"/home/neuro/Documents/BrainData/Bulk/GTEx/Formatted/GTEx-metadata-all.csv" ) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- hdbr = read.csv("/home/neuro/Documents/BrainData/Bulk/HDBR/Formatted/HDBR-metadata-subset.csv", header=TRUE, check.names = FALSE, row.names =1) -->
<!-- hdbr_decon = read.csv("/home/neuro/Documents/BrainData/Bulk/HDBR/Formatted/HDBR-decon.csv", header=TRUE, check.names = FALSE, row.names =1) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- hdbr_full = hdbr %>% -->
<!--   left_join(hdbr_decon  %>% -->
<!--               dplyr::select("SampleID", "Neurons", -->
<!--                             "Astrocytes", "Oligodendrocytes", -->
<!--                             "Microglia", "Endothelia"), by = "SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- write.csv(hdbr_full ,"/home/neuro/Documents/BrainData/Bulk/HDBR/Formatted/HDBR-metadata-all.csv" ) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pe = read.csv("/home/neuro/Documents/BrainData/Bulk/PsychEncode/Formatted/PsychEncode-metadata-subset.csv", header=TRUE, check.names = FALSE, row.names =1) -->
<!-- pe_decon = read.csv("/home/neuro/Documents/BrainData/Bulk/PsychEncode/Formatted/PsychEncode-decon.csv", header=TRUE, check.names = FALSE, row.names =1) %>% -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pe_full = pe %>% -->
<!--   left_join(pe_decon %>% -->
<!--               dplyr::select("SampleID", "Neurons", -->
<!--                             "Astrocytes", "Oligodendrocytes", -->
<!--                             "Microglia", "Endothelia"), by = "SampleID") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #write.csv(pe_full ,"/home/neuro/Documents/BrainData/Bulk/PsychEncode/Formatted/PsychEncode-metadata-all.csv" ) -->
<!-- ``` -->
</div>
</div>
<div id="export-data" class="section level1">
<h1>Export data</h1>
<pre class="r"><code>#write.csv(file = file.path(dir,&quot;Brainseq/Formatted/BrainSeq-metadata-subset.csv&quot;), md$BrainSeq)
#write.csv(file= file.path(dir,&quot;BrainSpan/Formatted/BrainSpan-metadata-subset.csv&quot;), md$BrainSpan)
#write.csv(file = file.path(dir,&quot;GTEx/Formatted/GTEx-metadata-subset.csv&quot;), md$GTEx)
#write.csv(file= file.path(dir, &quot;PsychEncode/Formatted/PsychEncode-metadata-subset.csv&quot;), md$PsychEncode)
#write.csv(file = file.path(dir, &quot;HDBR/Formatted/HDBR-metadata-subset.csv&quot;), md$HDBR)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
 [1] tools     grid      stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3        
 [3] ggpubr_0.6.0                pheatmap_1.0.12            
 [5] patchwork_1.2.0             UpSetR_1.4.0               
 [7] lubridate_1.9.3             forcats_1.0.0              
 [9] purrr_1.0.2                 readr_2.1.5                
[11] tidyverse_2.0.0             stringr_1.5.1              
[13] tidyr_1.3.0                 scales_1.3.0               
[15] data.table_1.14.10          readxl_1.4.3               
[17] naniar_1.0.0                glmpca_0.2.0               
[19] broom_1.0.5                 cowplot_1.1.2              
[21] glue_1.7.0                  ggfortify_0.4.16           
[23] stargazer_5.2.3             AnnotationHub_3.10.0       
[25] BiocFileCache_2.10.1        dbplyr_2.4.0               
[27] singscore_1.22.0            EnsDb.Hsapiens.v86_2.99.0  
[29] ensembldb_2.26.0            AnnotationFilter_1.26.0    
[31] GenomicFeatures_1.54.1      AnnotationDbi_1.64.1       
[33] DT_0.31                     variancePartition_1.32.2   
[35] BiocParallel_1.36.0         limma_3.58.1               
[37] gridExtra_2.3               pander_0.6.5               
[39] ggvenn_0.1.10               ggplot2_3.4.4              
[41] dplyr_1.1.4                 corrplot_0.92              
[43] reshape2_1.4.4              tibble_3.2.1               
[45] magrittr_2.0.3              recount3_1.12.0            
[47] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[49] GenomicRanges_1.54.1        GenomeInfoDb_1.38.5        
[51] IRanges_2.36.0              S4Vectors_0.40.2           
[53] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      
[55] matrixStats_1.2.0           workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] fs_1.6.3                      ProtGenerics_1.34.0          
  [3] bitops_1.0-7                  httr_1.4.7                   
  [5] RColorBrewer_1.1-3            numDeriv_2016.8-1.1          
  [7] backports_1.4.1               utf8_1.2.4                   
  [9] R6_2.5.1                      lazyeval_0.2.2               
 [11] withr_3.0.0                   prettyunits_1.2.0            
 [13] cli_3.6.2                     formatR_1.14                 
 [15] labeling_0.4.3                sass_0.4.8                   
 [17] mvtnorm_1.2-4                 Rsamtools_2.18.0             
 [19] R.utils_2.12.3                sessioninfo_1.2.2            
 [21] rstudioapi_0.15.0             RSQLite_2.3.5                
 [23] generics_0.1.3                BiocIO_1.12.0                
 [25] crosstalk_1.2.1               gtools_3.9.5                 
 [27] car_3.1-2                     Matrix_1.6-5                 
 [29] fansi_1.0.6                   abind_1.4-5                  
 [31] R.methodsS3_1.8.2             lifecycle_1.0.4              
 [33] whisker_0.4.1                 yaml_2.3.8                   
 [35] edgeR_4.0.11                  carData_3.0-5                
 [37] gplots_3.1.3                  SparseArray_1.2.3            
 [39] blob_1.2.4                    promises_1.2.1               
 [41] crayon_1.5.2                  lattice_0.22-5               
 [43] annotate_1.80.0               KEGGREST_1.42.0              
 [45] pillar_1.9.0                  knitr_1.45                   
 [47] rjson_0.2.21                  boot_1.3-28.1                
 [49] corpcor_1.6.10                codetools_0.2-19             
 [51] getPass_0.2-4                 vctrs_0.6.5                  
 [53] png_0.1-8                     Rdpack_2.6                   
 [55] cellranger_1.1.0              gtable_0.3.4                 
 [57] cachem_1.0.8                  xfun_0.41                    
 [59] rbibutils_2.2.16              S4Arrays_1.2.0               
 [61] mime_0.12                     iterators_1.0.14             
 [63] statmod_1.5.0                 interactiveDisplayBase_1.40.0
 [65] ellipsis_0.3.2                nlme_3.1-164                 
 [67] pbkrtest_0.5.2                bit64_4.0.5                  
 [69] progress_1.2.3                EnvStats_2.8.1               
 [71] filelock_1.0.3                rprojroot_2.0.4              
 [73] bslib_0.6.1                   KernSmooth_2.23-22           
 [75] colorspace_2.1-0              DBI_1.2.1                    
 [77] tidyselect_1.2.0              processx_3.8.3               
 [79] bit_4.0.5                     compiler_4.3.3               
 [81] curl_5.2.0                    git2r_0.33.0                 
 [83] graph_1.80.0                  xml2_1.3.6                   
 [85] DelayedArray_0.28.0           rtracklayer_1.62.0           
 [87] caTools_1.18.2                remaCor_0.0.16               
 [89] callr_3.7.3                   rappdirs_0.3.3               
 [91] digest_0.6.34                 minqa_1.2.6                  
 [93] rmarkdown_2.25                aod_1.3.3                    
 [95] XVector_0.42.0                RhpcBLASctl_0.23-42          
 [97] htmltools_0.5.7               pkgconfig_2.0.3              
 [99] lme4_1.1-35.1                 highr_0.10                   
[101] fastmap_1.1.1                 rlang_1.1.3                  
[103] htmlwidgets_1.6.4             shiny_1.8.0                  
[105] farver_2.1.1                  jquerylib_0.1.4              
[107] jsonlite_1.8.8                R.oo_1.25.0                  
[109] RCurl_1.98-1.14               GenomeInfoDbData_1.2.11      
[111] munsell_0.5.0                 Rcpp_1.0.12                  
[113] visdat_0.6.0                  stringi_1.8.3                
[115] zlibbioc_1.48.0               MASS_7.3-60.0.1              
[117] plyr_1.8.9                    parallel_4.3.3               
[119] Biostrings_2.70.1             splines_4.3.3                
[121] hms_1.1.3                     locfit_1.5-9.8               
[123] ps_1.7.6                      ggsignif_0.6.4               
[125] biomaRt_2.58.0                futile.options_1.0.1         
[127] BiocVersion_3.18.1            XML_3.99-0.16                
[129] evaluate_0.23                 lambda.r_1.2.4               
[131] BiocManager_1.30.22           nloptr_2.0.3                 
[133] tzdb_0.4.0                    httpuv_1.6.13                
[135] xtable_1.8-4                  restfulr_0.0.15              
[137] fANCOVA_0.6-1                 rstatix_0.7.2                
[139] later_1.3.2                   lmerTest_3.1-3               
[141] memoise_2.0.1                 GenomicAlignments_1.38.2     
[143] timechange_0.3.0              here_1.0.1                   
[145] GSEABase_1.64.0              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
