<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="unawaz1996" />

<meta name="date" content="2023-09-24" />

<title>bulk-deconvolution</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BITHub-preprocessing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proprocessing datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-bulk-preprocess-data.html">Bulk RNA-seq datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="03-metadata-attributes.html">Dataset characteristics</a>
</li>
<li>
  <a href="04-bulk-deconvolution.html">Deconvolution of bulk RNA-seq datasets</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/unawaz1996/BITHub-preprocessing">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">bulk-deconvolution</h1>
<h4 class="author">unawaz1996</h4>
<h4 class="date">2023-09-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>BITHub-preprocessing/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230919code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230919)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230919code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230919)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainspan-deconvolution-region.svg
</td>
<td style="text-align:left;">
output/Poster_plots/Brainspan-deconvolution-region.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainseq-deconvolution-region.svg
</td>
<td style="text-align:left;">
output/Poster_plots/Brainseq-deconvolution-region.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/GTEx-deconvolution-region.svg
</td>
<td style="text-align:left;">
output/Poster_plots/GTEx-deconvolution-region.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/HDBR-deconvolution-region.svg
</td>
<td style="text-align:left;">
output/Poster_plots/HDBR-deconvolution-region.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/PE-deconvolution-region.svg
</td>
<td style="text-align:left;">
output/Poster_plots/PE-deconvolution-region.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainspan-deconvolution-period.svg
</td>
<td style="text-align:left;">
output/Poster_plots/Brainspan-deconvolution-period.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainseq-deconvolution-period.svg
</td>
<td style="text-align:left;">
output/Poster_plots/Brainseq-deconvolution-period.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/GTEx-deconvolution-period.svg
</td>
<td style="text-align:left;">
output/Poster_plots/GTEx-deconvolution-period.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/HDBR-deconvolution-period.svg
</td>
<td style="text-align:left;">
output/Poster_plots/HDBR-deconvolution-period.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/PE-deconvolution-period.svg
</td>
<td style="text-align:left;">
output/Poster_plots/PE-deconvolution-period.svg
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996BITHubpreprocessingtreeddc9f1d9a827bc1628a3fc12e0b3c36447b24434targetblankddc9f1da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/tree/ddc9f1d9a827bc1628a3fc12e0b3c36447b24434" target="_blank">ddc9f1d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996BITHubpreprocessingtreeddc9f1d9a827bc1628a3fc12e0b3c36447b24434targetblankddc9f1da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/unawaz1996/BITHub-preprocessing/tree/ddc9f1d9a827bc1628a3fc12e0b3c36447b24434" target="_blank">ddc9f1d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    output/Thesis_plots/Variance/PCA/BrainSpan/brainspan-varPart-cor/

Untracked files:
    Untracked:  analysis/02-snRNAseq-preprocess.Rmd
    Untracked:  analysis/04-variation-drivers.Rmd
    Untracked:  output/Thesis_plots/Variance/PCA/BrainSeq/BrainSeq-PCA-mitoR.png
    Untracked:  output/Thesis_plots/Variance/PCA/BrainSpan/BrainSpan-PCA.png
    Untracked:  output/Thesis_plots/Variance/PCA/GTEx/GTEx-PCA-rRNA.png
    Untracked:  output/Thesis_plots/Variance/PCA/all-data.png

Unstaged changes:
    Deleted:    analysis/Metadata-attributes.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/bulk-deconvolution.Rmd
    Modified:   analysis/dataset-attributes.Rmd
    Modified:   analysis/snRNAseq-preprocess.Rmd
    Deleted:    analysis/variation-drives.Rmd
    Modified:   output/Thesis_plots/Attributes/Donor_and_sample/Aldringer_total.svg
    Modified:   output/Thesis_plots/Attributes/Donor_and_sample/PsychEncode_regions_total.svg
    Modified:   output/Thesis_plots/Metadata/BrainSeq-cor-seq-metadata.pdf
    Modified:   output/Thesis_plots/Metadata/GTEx-cor-seq-metadata.pdf
    Modified:   output/Thesis_plots/Metadata/HDBR-corplot.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/04-bulk-deconvolution.Rmd</code>)
and HTML (<code>docs/04-bulk-deconvolution.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/BITHub-preprocessing/blob/ddc9f1d9a827bc1628a3fc12e0b3c36447b24434/analysis/04-bulk-deconvolution.Rmd" target="_blank">ddc9f1d</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-04-02
</td>
<td>
wflow_publish(c("analysis/04-bulk-deconvolution.Rmd",
"analysis/03-metadata-attributes.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="background" class="section level1">
<h1>Background</h1>
<p>This R markdown document contains the complete code for the analysis
of deconvolved bulk RNA-seq expression profiles for Brain Integrative
Transcriptome Hub (BITHub). All datasets were prepossessed using the
code in <code>01-bulk-preprocess-data.Rmd</code>. The deconvolution
analysis itself was performed using <code>run-decon.R</code> code in
<code>/code/deconvolution/</code> sub directory of this project. The
script generates <code>decon-estimates.Rda</code> file which contains
deconvolution estimates for each bulk RNA-seq datasets.</p>
<p>In order to re-run the scripts - make sure to change the directories
to the location of where the input files for preprocessing are stored.
All directories are defined in each dataset section.</p>
<p>All other files are stored within the data folder of this R workflow
directory.</p>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="goodness-of-fit" class="section level2">
<h2>Goodness of fit</h2>
<p>CIBERSORT is used for the analysis, and MultiBrain signatures
generated in Sutton et al (2022) are used for signatures. The
performance of the deconvolution analysis was evaluated using a
goodness-of-fit (r) measurement, which is based on the Pearson’s
correlation coefficient between the expression data and measured
expression data.</p>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-4-2.png" width="960" style="display: block; margin: auto;" /><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-4-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.4825263</code></pre>
<pre><code>[1] 0.6266852</code></pre>
</div>
<div id="cell-type-proportion-trends" class="section level2">
<h2>Cell-type proportion trends</h2>
<div id="across-regions" class="section level3">
<h3>Across regions</h3>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="across-age-intervals-and-regions" class="section level3">
<h3>Across age intervals and regions</h3>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-8-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-9-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-10-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-11-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-12-1.png" width="1920" style="display: block; margin: auto;" /></p>
<!-- - Across age intervals and regions  -->
<!-- ```{r} -->
<!-- bseq.cbs %>%  -->
<!--   melt() %>% -->
<!--   mutate(AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!--                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!--                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!--                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!--                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!--                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!--                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>% -->
<!--   ggplot(aes(x = AgeInterval,y = value, fill = Regions)) + geom_boxplot() + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10, angle = 90), -->
<!--         axis.text.y = element_text(size = 12),  -->
<!--         legend.position = "none") + facet_grid(~variable)  -->
<!-- ``` -->
<p><img src="figure/04-bulk-deconvolution.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ```{r fig.height=12, fig.width=20} -->
<!--  rbind(pe.cbs, -->
<!--       bspan.csb, -->
<!--       bseq.cbs,  -->
<!--       gtex.cbs,  -->
<!--       hdbr.cbs) %>%  -->
<!--   as.data.frame %>% -->
<!--   melt() %>% -->
<!--   drop_na(Regions) %>% -->
<!--  # dplyr::filter(Dataset != "HDBR") %>% -->
<!--   mutate(variable= factor(variable, levels = c("Neurons","Astrocytes",  -->
<!--                                                "Endothelia", "Oligodendrocytes", -->
<!--                                                "Microglia" -->
<!--                                               )), -->
<!--         Regions = factor(Regions,  -->
<!--                          levels = c("Brain", "Forebrain","Hindbrain", -->
<!--                                     "Midbrain", "Forebrain and midbrain",  -->
<!--                                     "Cortex", "Subcortex", -->
<!--                                     "Cerebellum", "Spinal Cord")), -->
<!--         AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!--                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!--                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!--                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!--                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!--                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!--                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>%  -->
<!--   dplyr::filter(variable == "Neurons") %>% -->
<!--   ggplot(aes(x = AgeInterval, y = value, fill = Regions)) + geom_boxplot() + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 12),  -->
<!--         legend.position = "none") +  -->
<!--   facet_grid(Dataset~variable, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Forebrain and midbrain" = "#9D696C" -->
<!--                                 )) -->
<!-- ``` -->
<!-- # ```{r fig.height=12, fig.width=20} -->
<!-- #  rbind(pe.cbs, -->
<!-- #       bspan.csb, -->
<!-- #       bseq.cbs,  -->
<!-- #       gtex.cbs,  -->
<!-- #       hdbr.cbs) %>%  -->
<!-- #   as.data.frame %>% -->
<!-- #   melt() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #  # dplyr::filter(Dataset != "HDBR") %>% -->
<!-- #   mutate(variable= factor(variable, levels = c("Neurons","Astrocytes",  -->
<!-- #                                                "Endothelia", "Oligodendrocytes", -->
<!-- #                                                "Microglia" -->
<!-- #                                                -->
<!-- #                                               )), -->
<!-- #         Regions = factor(Regions,  -->
<!-- #                          levels = c("Brain", "Forebrain","Hindbrain", -->
<!-- #                                     "Midbrain", "Forebrain and midbrain",  -->
<!-- #                                     "Cortex", "Subcortex", -->
<!-- #                                     "Cerebellum", "Spinal Cord")), -->
<!-- #         AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!-- #                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!-- #                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!-- #                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!-- #                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!-- #                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!-- #                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>%  -->
<!-- #   dplyr::filter(variable == "Astrocytes") %>% -->
<!-- #   ggplot(aes(x = AgeInterval, y = value, fill = Regions)) + geom_boxplot() + -->
<!-- #   theme_bw() + -->
<!-- #   theme(axis.text.x = element_text(size = 10), -->
<!-- #         axis.text.y = element_text(size = 12),  -->
<!-- #         legend.position = "none") +  -->
<!-- #   facet_grid(Dataset~variable, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!-- #   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!-- #                                  "Subcortex" = "#838DE0",  -->
<!-- #                                  "Cerebellum" = "#FFD256", -->
<!-- #                                  "Spinal Cord" = "#C6D14A",  -->
<!-- #                                  "Forebrain" = "#F6B19D", # -->
<!-- #                                  "Midbrain" = "#AC93AD", -->
<!-- #                                  "Hindbrain" = "#6E8B8E", -->
<!-- #                                  "Brain" = "#FFA6B7",  -->
<!-- #                                  "Forebrain and midbrain" = "#9D696C" -->
<!-- #                                 )) -->
<!-- #  -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- #  -->
<!-- # ```{r fig.height=12, fig.width=20} -->
<!-- #  rbind(pe.cbs, -->
<!-- #       bspan.csb, -->
<!-- #       bseq.cbs,  -->
<!-- #       gtex.cbs,  -->
<!-- #       hdbr.cbs) %>%  -->
<!-- #   as.data.frame %>% -->
<!-- #   melt() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #  # dplyr::filter(Dataset != "HDBR") %>% -->
<!-- #   mutate(variable= factor(variable, levels = c("Neurons","Astrocytes",  -->
<!-- #                                                "Endothelia", "Oligodendrocytes", -->
<!-- #                                                "Microglia" -->
<!-- #                                                -->
<!-- #                                               )), -->
<!-- #         Regions = factor(Regions,  -->
<!-- #                          levels = c("Brain", "Forebrain","Hindbrain", -->
<!-- #                                     "Midbrain", "Forebrain and midbrain",  -->
<!-- #                                     "Cortex", "Subcortex", -->
<!-- #                                     "Cerebellum", "Spinal Cord")), -->
<!-- #         AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!-- #                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!-- #                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!-- #                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!-- #                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!-- #                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!-- #                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>%  -->
<!-- #   dplyr::filter(variable == "Microglia") %>% -->
<!-- #   ggplot(aes(x = AgeInterval, y = value, fill = Regions)) + geom_boxplot() + -->
<!-- #   theme_bw() + -->
<!-- #   theme(axis.text.x = element_text(size = 10), -->
<!-- #         axis.text.y = element_text(size = 12),  -->
<!-- #         legend.position = "none") +  -->
<!-- #   facet_grid(Dataset~variable, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!-- #   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!-- #                                  "Subcortex" = "#838DE0",  -->
<!-- #                                  "Cerebellum" = "#FFD256", -->
<!-- #                                  "Spinal Cord" = "#C6D14A",  -->
<!-- #                                  "Forebrain" = "#F6B19D", # -->
<!-- #                                  "Midbrain" = "#AC93AD", -->
<!-- #                                  "Hindbrain" = "#6E8B8E", -->
<!-- #                                  "Brain" = "#FFA6B7",  -->
<!-- #                                  "Forebrain and midbrain" = "#9D696C" -->
<!-- #                                 )) -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r fig.height=12, fig.width=20} -->
<!-- #  rbind(pe.cbs, -->
<!-- #       bspan.csb, -->
<!-- #       bseq.cbs,  -->
<!-- #       gtex.cbs,  -->
<!-- #       hdbr.cbs) %>%  -->
<!-- #   as.data.frame %>% -->
<!-- #   melt() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #  # dplyr::filter(Dataset != "HDBR") %>% -->
<!-- #   mutate(variable= factor(variable, levels = c("Neurons","Astrocytes",  -->
<!-- #                                                "Endothelia", "Oligodendrocytes", -->
<!-- #                                                "Microglia" -->
<!-- #                                                -->
<!-- #                                               )), -->
<!-- #         Regions = factor(Regions,  -->
<!-- #                          levels = c("Brain", "Forebrain","Hindbrain", -->
<!-- #                                     "Midbrain", "Forebrain and midbrain",  -->
<!-- #                                     "Cortex", "Subcortex", -->
<!-- #                                     "Cerebellum", "Spinal Cord")), -->
<!-- #         AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!-- #                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!-- #                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!-- #                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!-- #                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!-- #                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!-- #                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>%  -->
<!-- #   dplyr::filter(variable == "Endothelia") %>% -->
<!-- #   ggplot(aes(x = AgeInterval, y = value, fill = Regions)) + geom_boxplot() + -->
<!-- #   theme_bw() + -->
<!-- #   theme(axis.text.x = element_text(size = 10), -->
<!-- #         axis.text.y = element_text(size = 12),  -->
<!-- #         legend.position = "none") +  -->
<!-- #   facet_grid(Dataset~variable, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!-- #   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!-- #                                  "Subcortex" = "#838DE0",  -->
<!-- #                                  "Cerebellum" = "#FFD256", -->
<!-- #                                  "Spinal Cord" = "#C6D14A",  -->
<!-- #                                  "Forebrain" = "#F6B19D", # -->
<!-- #                                  "Midbrain" = "#AC93AD", -->
<!-- #                                  "Hindbrain" = "#6E8B8E", -->
<!-- #                                  "Brain" = "#FFA6B7",  -->
<!-- #                                  "Forebrain and midbrain" = "#9D696C" -->
<!-- #                                 )) -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r fig.height=12, fig.width=20} -->
<!-- #  rbind(pe.cbs, -->
<!-- #       bspan.csb, -->
<!-- #       bseq.cbs,  -->
<!-- #       gtex.cbs,  -->
<!-- #       hdbr.cbs) %>%  -->
<!-- #   as.data.frame %>% -->
<!-- #   melt() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #  # dplyr::filter(Dataset != "HDBR") %>% -->
<!-- #   mutate(variable= factor(variable, levels = c("Neurons","Astrocytes",  -->
<!-- #                                                "Endothelia", "Oligodendrocytes", -->
<!-- #                                                "Microglia" -->
<!-- #                                                -->
<!-- #                                               )), -->
<!-- #         Regions = factor(Regions,  -->
<!-- #                          levels = c("Brain", "Forebrain","Hindbrain", -->
<!-- #                                     "Midbrain", "Forebrain and midbrain",  -->
<!-- #                                     "Cortex", "Subcortex", -->
<!-- #                                     "Cerebellum", "Spinal Cord")), -->
<!-- #         AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw", -->
<!-- #                                                         "10-12pcw", "13-15pcw", "16-18pcw", -->
<!-- #                                                         "19-24pcw", "25-38pcw", "0-5mos", -->
<!-- #                                                         "6-18mos", "19mos-5yrs", "6-11yrs", -->
<!-- #                                                         "12-19yrs", "20-29yrs", "30-39yrs", -->
<!-- #                                                       "40-49yrs", "50-59yrs", "60-69yrs", -->
<!-- #                                                       "70-79yrs", "80-89yrs", "90-99yrs"))) %>%  -->
<!-- #   dplyr::filter(variable == "Oligodendrocytes") %>% -->
<!-- #   ggplot(aes(x = AgeInterval, y = value, fill = Regions)) + geom_boxplot() + -->
<!-- #   theme_bw() + -->
<!-- #   theme(axis.text.x = element_text(size = 10), -->
<!-- #         axis.text.y = element_text(size = 12),  -->
<!-- #         legend.position = "none") +  -->
<!-- #   facet_grid(Dataset~variable, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!-- #   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!-- #                                  "Subcortex" = "#838DE0",  -->
<!-- #                                  "Cerebellum" = "#FFD256", -->
<!-- #                                  "Spinal Cord" = "#C6D14A",  -->
<!-- #                                  "Forebrain" = "#F6B19D", # -->
<!-- #                                  "Midbrain" = "#AC93AD", -->
<!-- #                                  "Hindbrain" = "#6E8B8E", -->
<!-- #                                  "Brain" = "#FFA6B7",  -->
<!-- #                                  "Forebrain and midbrain" = "#9D696C" -->
<!-- #                                 )) -->
<!-- #  -->
<!-- # ``` -->
<!-- # ```{r} -->
<!-- # bseq.cbs %>%  -->
<!-- #   dplyr::select(Regions, Neurons, Astrocytes, Oligodendrocytes,Microglia, Endothelia ) %>% -->
<!-- #   group_by(Regions) %>% dplyr::summarise(across(everything(), list(mean)) * 100) -->
<!-- #  -->
<!-- #  -->
<!-- # gtex.cbs %>%  -->
<!-- #   dplyr::select(Regions, Neurons, Astrocytes, Oligodendrocytes,Microglia, Endothelia ) %>% -->
<!-- #   group_by(Regions) %>% dplyr::summarise(across(everything(), list(mean)) * 100) -->
<!-- # ``` -->
<!-- ## Impact of cell fractions on gene expression  -->
<!-- ```{r} -->
<!-- thresh <- function(x) { -->
<!--       y <- x > 1 -->
<!--       keep <- rowSums(y) > (ncol(y) / 10) -->
<!--       return(x[keep,]) -->
<!--     } -->
<!-- ``` -->
<!-- ```{r} -->
<!-- gtex <- thresh(e$GTEx) -->
<!-- span <- thresh(e$BSpan) -->
<!-- seq <- thresh(e$BSeq) -->
<!-- pe <- thresh(e$PE) -->
<!-- hdbr <- thresh(e$HDBR) -->
<!-- ``` -->
<!-- - BrainSeq -->
<!-- ```{r} -->
<!-- pc = log(seq + 0.05) %>%  -->
<!--   t() %>%  -->
<!--    prcomp(scale = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!--  pc$x  %>% -->
<!--     as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   left_join(bseq.cbs) %>% -->
<!--   left_join(md$BSeq %>%  -->
<!--               dplyr::select(SampleID, DonorID,  -->
<!--                             Sex, Ethnicity,  -->
<!--                             Diagnosis, Period)) %>% -->
<!--     dplyr::select( -->
<!--         PC1, PC2, PC3, -->
<!--         Neurons, -->
<!--         Astrocytes, -->
<!--         Oligodendrocytes, -->
<!--          Microglia, -->
<!--         Endothelia, -->
<!--         Period,  -->
<!--         Regions, -->
<!--         DonorID,  -->
<!--         Diagnosis -->
<!--     ) %>%  -->
<!--   drop_na(Regions) %>% -->
<!--    mutate(Period = as.numeric(as.factor(Period)),  -->
<!--           Regions = as.numeric(as.factor(Regions)), -->
<!--             DonorID = as.numeric(as.factor(DonorID)),  -->
<!--           Diagnosis = as.numeric(as.factor(Diagnosis))) %>% -->
<!--     cor() %>%  -->
<!--     corrplot( -->
<!--         type = "lower",  -->
<!--         diag = FALSE,  -->
<!--         addCoef.col = 1,  -->
<!--         number.cex = 0.75) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- bseq.cbs %>%  -->
<!--   dplyr::select(SampleID, Neurons, -->
<!--         Astrocytes, -->
<!--         Oligodendrocytes, -->
<!--          Microglia, -->
<!--         Endothelia) %>% -->
<!--   left_join(md$BSeq %>%  -->
<!--               dplyr::select(SampleID,"Dev.Replicating", "Dev.Quiescent","Adult.Neurons", -->
<!--                             "Adult.Astrocytes", "Adult.Oligo", -->
<!--                             "Adult.Microglia")) %>% -->
<!--   dplyr::select(-SampleID) %>% -->
<!--    cor() %>%  -->
<!--     corrplot( -->
<!--         type = "lower",  -->
<!--         diag = FALSE,  -->
<!--         addCoef.col = 1,  -->
<!--         number.cex = 0.75) -->
<!-- ``` -->
<!-- - BrainSpan -->
<!-- ```{r} -->
<!-- pc = log(span + 0.05) %>%  -->
<!--   t() %>%  -->
<!--    prcomp(scale = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!--  cor.bspan =pc$x  %>% -->
<!--     as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   left_join(bspan.csb) %>% -->
<!--   left_join(md$BSpan %>%  -->
<!--               dplyr::select(SampleID, DonorID,  -->
<!--                             Sex, Ethnicity,  -->
<!--                             StructureAcronym, Period)) %>% -->
<!--     dplyr::select( -->
<!--         PC1, PC2, PC3, -->
<!--         Neurons, -->
<!--         Astrocytes, -->
<!--         Oligodendrocytes, -->
<!--          Microglia, -->
<!--         Endothelia, -->
<!--         Period,  -->
<!--         Regions, -->
<!--         DonorID,  -->
<!--         StructureAcronym -->
<!--     ) %>%  -->
<!--   drop_na(Regions) %>% -->
<!--    mutate(Period = as.numeric(as.factor(Period)),  -->
<!--           Regions = as.numeric(as.factor(Regions)), -->
<!--             DonorID = as.numeric(as.factor(DonorID)),  -->
<!--           StructureAcronym = as.numeric(as.factor(StructureAcronym))) %>% -->
<!--     cor()  -->
<!-- cor.bspan = cor.bspan[-c(1:3),] -->
<!-- cor.bspan = cor.bspan[,c(1:3)] -->
<!-- cor.bspan %>% corrplot( -->
<!--         addCoef.col = 1,  -->
<!--         number.cex = 0.75,  -->
<!--         method = 'color') -->
<!-- ``` -->
<!-- - GTEx -->
<!-- ```{r} -->
<!-- pc = log(gtex + 0.05) %>%  -->
<!--   t() %>%  -->
<!--    prcomp(scale = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!--  cor.gtex =pc$x  %>% -->
<!--     as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   left_join(gtex.cbs) %>% -->
<!--   left_join(md$GTEx %>%  -->
<!--               dplyr::select(SampleID, DonorID,  -->
<!--                             Sex, -->
<!--                             StructureAcronym, Period, rRNA_ratio, PMI)) %>% -->
<!--     dplyr::select( -->
<!--         PC1, PC2, PC3, -->
<!--         Neurons, -->
<!--         Astrocytes, -->
<!--         Oligodendrocytes, -->
<!--          Microglia, -->
<!--         Endothelia, -->
<!--         Regions, -->
<!--         DonorID,  -->
<!--         StructureAcronym,  -->
<!--         rRNA_ratio,  -->
<!--         PMI -->
<!--     ) %>%  -->
<!--   drop_na(Regions) %>% -->
<!--    mutate( -->
<!--           Regions = as.numeric(as.factor(Regions)), -->
<!--             DonorID = as.numeric(as.factor(DonorID)),  -->
<!--           StructureAcronym = as.numeric(as.factor(StructureAcronym))) %>% -->
<!--     cor()  -->
<!-- cor.gtex = cor.gtex[-c(1:3),] -->
<!-- cor.gtex = cor.gtex[,c(1:3)] -->
<!-- cor.gtex %>% corrplot( -->
<!--         addCoef.col = 1,  -->
<!--         number.cex = 0.75,  -->
<!--         method = 'color') -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pc = log(pe + 0.05) %>%  -->
<!--   t() %>%  -->
<!--    prcomp(scale = TRUE) -->
<!-- ``` -->
<!-- ```{r} -->
<!--  pc$x  %>% -->
<!--     as.data.frame() %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   left_join(pe.cbs) %>% -->
<!--   left_join(md$PE %>%  -->
<!--               dplyr::select(SampleID, Diagnosis, AgeNumeric)) %>% -->
<!--     dplyr::select( -->
<!--         PC1, PC2, PC3, -->
<!--         Neurons, -->
<!--         Astrocytes, -->
<!--         Oligodendrocytes, -->
<!--          Microglia, -->
<!--         Endothelia, -->
<!--         Period, -->
<!--         Diagnosis -->
<!--     ) %>% -->
<!--    mutate(Period = as.numeric(as.factor(Period)), -->
<!--             Diagnosis = as.numeric(as.factor(Diagnosis))) %>% -->
<!--     cor() %>%  -->
<!--     corrplot( -->
<!--         type = "lower",  -->
<!--         diag = FALSE,  -->
<!--         addCoef.col = 1,  -->
<!--         number.cex = 0.75) -->
<!-- ``` -->
<!-- ## Export the datasets  -->
<!-- ## GOF poster  -->
<!-- ```{r} -->
<!-- library(ggstatsplot) -->
<!-- brainSeq_region = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "BrainSeq") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Regions, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Forebrain and midbrain" = "#9D696C" -->
<!--                                 )) +  -->
<!--   ggtitle("BrainSeq") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainSpan_region = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "BrainSpan") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Regions, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Forebrain and midbrain" = "#9D696C" -->
<!--                                 )) +  -->
<!--   ggtitle("BrainSpan") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- gtex_region = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "GTEx") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Regions, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Forebrain and midbrain" = "#9D696C" -->
<!--                                 )) +  -->
<!--   ggtitle("GTEx") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- hdbr_region = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "HDBR") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Regions, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Chroid plexus" = "#9D696C" -->
<!--                                 )) +  -->
<!--   ggtitle("HDBR") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pe_region = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "PsychEncode") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Regions, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Cortex" = "#E85571",  -->
<!--                                  "Subcortex" = "#838DE0",  -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A",  -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7",  -->
<!--                                  "Chroid plexus" = "#9D696C" -->
<!--                                 )) +  -->
<!--   ggtitle("PE") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ggsave(brainSpan_region, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainspan-deconvolution-region.svg", height = 7, width = 8, units = "in") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ggsave(brainSeq_region, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainseq-deconvolution-region.svg", height = 7, width = 8, units = "in") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ggsave(gtex_region, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/GTEx-deconvolution-region.svg", height = 7, width = 8, units = "in") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ggsave(hdbr_region, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/HDBR-deconvolution-region.svg", height = 7, width = 8, units = "in") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ggsave(pe_region, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/PE-deconvolution-region.svg", height = 7, width = 8, units = "in") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- library(ggstatsplot) -->
<!-- brainSeq_period = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "BrainSeq") %>%  -->
<!--   mutate(Period = factor(Period, levels = c("Prenatal", "Postnatal"))) %>% -->
<!--   ggbetweenstats( -->
<!--   x     = Period, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!-- scale_color_manual(values = c("Prenatal" = "#FD84B8",  -->
<!--                                                 "Postnatal" = "#3A53A4")) +  -->
<!--   ggtitle("BrainSeq") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- brainSpan_period = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "BrainSpan") %>%  -->
<!--     mutate(Period = factor(Period, levels = c("Prenatal", "Postnatal"))) %>% -->
<!--   ggbetweenstats( -->
<!--   x     = Period, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!-- scale_color_manual(values = c("Prenatal" = "#FD84B8",  -->
<!--                                                 "Postnatal" = "#3A53A4")) +  -->
<!--   ggtitle("BrainSpan") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- gtex_period = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "GTEx") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Period, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!-- scale_color_manual(values = c("Prenatal" = "#FD84B8",  -->
<!--                                                 "Postnatal" = "#3A53A4")) +  -->
<!--   ggtitle("GTEx") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- hdbr_period = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "HDBR") %>%  -->
<!--   ggbetweenstats( -->
<!--   x     = Period, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Prenatal" = "#FD84B8",  -->
<!--                                                 "Postnatal" = "#3A53A4")) +  -->
<!--   ggtitle("HDBR") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pe_period = do.call(rbind, gof_results) %>% -->
<!--   as.data.frame() %>% -->
<!--   drop_na(Regions) %>% -->
<!--   dplyr::filter(Deconvolution == "Cibersort" & Dataset == "PsychEncode") %>%  -->
<!--     mutate(Period = factor(Period, levels = c("Prenatal", "Postnatal"))) %>% -->
<!--   ggbetweenstats( -->
<!--   x     = Period, -->
<!--   y     = r,  -->
<!--   package = "RColorBrewer", -->
<!--   palette = "Pastel1" -->
<!-- ) + theme_bw() + geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--    ylab("Goodness of fit (r)") + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"), -->
<!--         axis.text.y = element_text(size = 12, family = "serif", color = "black"),  -->
<!--         legend.position = "none",  -->
<!--         strip.text = element_text(size = 12,family = "serif"),  -->
<!--         rect = element_rect(fill = "transparent")) +  -->
<!--   scale_color_manual(values = c("Prenatal" = "#FD84B8",  -->
<!--                                                 "Postnatal" = "#3A53A4"))  +  -->
<!--   ggtitle("PE") -->
<!-- ``` -->
</div>
</div>
<div id="export" class="section level2">
<h2>Export</h2>
<div id="datasets" class="section level3">
<h3>Datasets</h3>
</div>
<div id="figures" class="section level3">
<h3>Figures</h3>
<!-- # ```{r} -->
<!-- # ggsave(brainSpan_period, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainspan-deconvolution-period.svg", height = 7, width = 8, units = "in") -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # ggsave(brainSeq_period, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/Brainseq-deconvolution-period.svg", height = 7, width = 8, units = "in") -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # ggsave(gtex_period, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/GTEx-deconvolution-period.svg", height = 7, width = 8, units = "in") -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # ggsave(hdbr_period, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/HDBR-deconvolution-period.svg", height = 7, width = 8, units = "in") -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # ggsave(pe_period, filename = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Poster_plots/PE-deconvolution-period.svg", height = 7, width = 8, units = "in") -->
<!-- #  -->
<!-- # ``` -->
<!-- ## Goodness-of-fit -->
<!-- # ```{r} -->
<!-- # ## BrainSeq -->
<!-- # gof_results$BSeq_DRS = write.gof(e$BSeq, decon_results$BSeq_DRS, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "BrainSeq", -->
<!-- #                              Deconvolution = "DeconRNASeq") %>% -->
<!-- #   rownames_to_column("SampleID") %>% -->
<!-- #   left_join(md$BSeq %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- # gof_results$BSeq_DTG = write.gof(e$BSeq, decon_results$BSeq_DTG, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "BrainSeq", -->
<!-- #                              Deconvolution = "Dtangle") %>% -->
<!-- #   rownames_to_column("SampleID")  %>% -->
<!-- #   left_join(md$BSeq %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # ## GTEx -->
<!-- # gof_results$GTEx_DRS = write.gof(e$GTEx, decon_results$GTEx_DRS, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "GTEx", -->
<!-- #                              Deconvolution = "DeconRNASeq")%>% -->
<!-- #   rownames_to_column("SampleID") %>% -->
<!-- #   mutate(SampleID = gsub("\\.", "-", SampleID))  %>% -->
<!-- #   left_join(md$GTEx %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- # gof_results$GTEx_DTG = write.gof(e$GTEx, decon_results$GTEx_DTG, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "GTEx", -->
<!-- #                              Deconvolution = "Dtangle")%>% -->
<!-- #   rownames_to_column("SampleID") %>% -->
<!-- #   mutate(SampleID = gsub("\\.", "-", SampleID)) %>% -->
<!-- #   left_join(md$GTEx %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # ## PsychEncode -->
<!-- # gof_results$PE_DRS = write.gof(e$PE, decon_results$PE_DRS, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "PsychEncode", -->
<!-- #                              Deconvolution = "DeconRNASeq") %>% -->
<!-- #   rownames_to_column("SampleID") %>% -->
<!-- #   mutate(SampleID = gsub("X", "", SampleID), -->
<!-- #          SampleID = gsub("\\.", "-", SampleID)) %>% -->
<!-- #   left_join(md$PE %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- # gof_results$PE_DTG = write.gof(e$PE, decon_results$PE_DTG, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "PsychEncode", -->
<!-- #                              Deconvolution = "Dtangle") %>% -->
<!-- #   rownames_to_column("SampleID") %>% -->
<!-- #   mutate(SampleID = gsub("X", "", SampleID), -->
<!-- #          SampleID = gsub("\\.", "-", SampleID)) %>% -->
<!-- #   left_join(md$PE %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # ## HDBR -->
<!-- # gof_results$HDBR_DRS = write.gof(e$HDBR, decon_results$HDBR_DRS, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "HDBR", -->
<!-- #                              Deconvolution = "DeconRNASeq") %>% -->
<!-- #   rownames_to_column("SampleID")  %>% -->
<!-- #   left_join(md$HDBR %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- # gof_results$HDBR_DTG = write.gof(e$HDBR, decon_results$HDBR_DTG, sigsBrain$MB) %>% -->
<!-- #   as.data.frame() %>% mutate(Dataset= "HDBR", -->
<!-- #                              Deconvolution = "Dtangle") %>% -->
<!-- #   rownames_to_column("SampleID")  %>% -->
<!-- #   left_join(md$HDBR %>% -->
<!-- #               dplyr::select(SampleID, Regions, Period), by = "SampleID") -->
<!-- #  -->
<!-- #  -->
<!-- # ``` -->
<!-- # ```{r} -->
<!-- # cbs_regions = do.call(rbind, gof_results) %>% -->
<!-- #   as.data.frame() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #   dplyr::filter(Deconvolution == "Cibersort") %>%  -->
<!-- #     ggplot(aes(Regions, r, fill = Regions)) + geom_violin(width=1, alpha = 0.85) + -->
<!-- #       geom_boxplot(width=0.1, color="white") + -->
<!-- #       theme_bw() +   scale_fill_manual(values = c("Cortex" = "#E85571",  -->
<!-- #                                  "Subcortex" = "#838DE0",  -->
<!-- #                                  "Cerebellum" = "#FFD256", -->
<!-- #                                  "Spinal Cord" = "#C6D14A",  -->
<!-- #                                  "Forebrain" = "#F6B19D", # -->
<!-- #                                  "Midbrain" = "#AC93AD", -->
<!-- #                                  "Hindbrain" = "#6E8B8E", -->
<!-- #                                  "Brain" = "#FFA6B7",  -->
<!-- #                                  "Forebrain and midbrain" = "#9D696C" -->
<!-- #                                 )) +  -->
<!-- #   geom_hline(yintercept = c(0.5,0.7), color = "grey",linetype='dotted') +  -->
<!-- #   ylab("Goodness of fit (r)") + facet_wrap(~Dataset, scales = "free",  nrow = 6) +  -->
<!-- #   theme(legend.position = "none",  -->
<!-- #         axis.text = element_text(size =10), -->
<!-- #         axis.title  = element_text(size =12)) -->
<!-- #  -->
<!-- #  -->
<!-- # cbs_period =do.call(rbind, gof_results) %>% -->
<!-- #   as.data.frame() %>% -->
<!-- #   drop_na(Regions) %>% -->
<!-- #   mutate(Period = factor(Period, levels=c("Prenatal", "Postnatal"))) %>%  -->
<!-- #   dplyr::filter(Deconvolution == "Cibersort") %>%  -->
<!-- #     ggplot(aes(Period, r, fill = Period)) + geom_violin(width=1, alpha = 0.85) + -->
<!-- #       geom_boxplot(width=0.1, color="white") + -->
<!-- #       theme_bw() +  scale_fill_manual(values = c("Prenatal" = "#FD84B8",  -->
<!-- #                                                 "Postnatal" = "#54B4E6")) +  -->
<!-- #   geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') +  -->
<!-- #   ylab("Goodness of fit (r)") + facet_wrap(~Dataset, scales = "free",  nrow = 6) + -->
<!-- #   theme(legend.position = "none",  -->
<!-- #         axis.text = element_text(size =10), -->
<!-- #         axis.title  = element_text(size =12))  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # grid.arrange(cbs_regions, cbs_period, ncol=2) -->
<!-- # ``` -->
<!-- - Overall -->
<!-- ```{r} -->
<!-- overall = do.call(rbind, gof_results) %>% -->
<!--   ggplot(aes(y = r, x = Deconvolution, fill = Deconvolution)) + -->
<!--   geom_violin( alpha = 0.75) + -->
<!--       geom_boxplot(width=0.1)+ facet_grid(~Dataset) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_blank(), -->
<!--         legend.position = "top", -->
<!--         axis.title.x = element_blank(), -->
<!--         axis.text.y = element_text(size = 12), -->
<!--         axis.title.y = element_text(size = 15), -->
<!--         panel.grid.major = element_blank(), -->
<!--         panel.grid.minor = element_blank()) + -->
<!--   geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--   ylab("Goodness of fit (r)") + -->
<!--   scale_fill_manual(values = -->
<!--                       c("Cibersort" = "#fdaf91", -->
<!--                         "DeconRNASeq" = "#ad002b", -->
<!--                         "Dtangle" = "#925e9f")) -->
<!-- overall -->
<!-- ggsave(plot = overall, -->
<!--        filename = here::here("output/Thesis_plots/Deconvolution/GoF-overall.svg"), -->
<!--        height = 4, -->
<!--        width = 7.31, -->
<!--        units = "in") -->
<!-- ``` -->
<!-- - Across developmental periods -->
<!-- ```{r} -->
<!-- period_gof = do.call(rbind, gof_results) %>% -->
<!--   drop_na(Period) %>% -->
<!--   mutate(Period = factor(Period, levels = c("Prenatal", "Postnatal"))) %>% -->
<!--   ggplot(aes(y = r, x = Deconvolution, fill = Deconvolution)) + -->
<!--   geom_violin( alpha = 0.75) + -->
<!--       geom_boxplot(width=0.1)+ facet_grid(Dataset~Period) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_blank(), -->
<!--         legend.position = "top", -->
<!--         axis.title.x = element_blank(), -->
<!--         axis.text.y = element_text(size = 12), -->
<!--         axis.title.y = element_text(size = 15), -->
<!--         panel.grid.major = element_blank(), -->
<!--         panel.grid.minor = element_blank()) + -->
<!--   geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--   ylab("Goodness of fit (r)") + -->
<!--   scale_fill_manual(values = -->
<!--                       c("Cibersort" = "#fdaf91", -->
<!--                         "DeconRNASeq" = "#ad002b", -->
<!--                         "Dtangle" = "#925e9f")) -->
<!-- period_gof -->
<!-- ggsave(plot = period_gof, -->
<!--        filename = here::here("output/Thesis_plots/Deconvolution/GoF-period.svg"), -->
<!--        height = 9.30, -->
<!--        width = 8.15, -->
<!--        units = "in") -->
<!-- ``` -->
<!-- - Across regions -->
<!-- ```{r} -->
<!-- dodge <- position_dodge(width = 0.8) -->
<!-- regions = do.call(rbind, gof_results) %>% -->
<!--   drop_na(Period) %>% -->
<!--   ggplot(aes(y = r, x = Deconvolution, fill = Regions)) + -->
<!--   geom_boxplot() + -->
<!--  # -->
<!--   facet_grid(Dataset~Deconvolution, scales = "free_x") + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_blank(), -->
<!--         legend.position = "top", -->
<!--         axis.title.x = element_blank(), -->
<!--         axis.text.y = element_text(size = 12), -->
<!--         axis.title.y = element_text(size = 15), -->
<!--         panel.grid.major = element_blank(), -->
<!--         panel.grid.minor = element_blank()) + -->
<!--   geom_hline(yintercept = c(0.5,0.7), color = "black",linetype='dotted') + -->
<!--   ylab("Goodness of fit (r)") + -->
<!--     scale_fill_manual(values = c("Cortex" = "#E85571", -->
<!--                                  "Subcortex" = "#838DE0", -->
<!--                                  "Cerebellum" = "#FFD256", -->
<!--                                  "Spinal Cord" = "#C6D14A", -->
<!--                                  "Forebrain" = "#F6B19D", # -->
<!--                                  "Midbrain" = "#AC93AD", -->
<!--                                  "Hindbrain" = "#6E8B8E", -->
<!--                                  "Brain" = "#FFA6B7", -->
<!--                                  "Forebrain and midbrain" = "#9D696C" -->
<!--                                 )) -->
<!-- ggsave(plot = regions, -->
<!--        filename = here::here("output/Thesis_plots/Deconvolution/GoF-regions.svg"), -->
<!--        height = 9.30, -->
<!--        width = 7.31, -->
<!--        units = "in") -->
<!-- ``` -->
<!-- ## Proportion estimation -->
<!-- - BrainSeq -->
<!-- ```{r} -->
<!-- decon_bseq =decon_results$BSeq_CIBERSORT %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSeq) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$BSeq_DRS %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSeq) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$BSeq_DTG %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSeq) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- - BrainSpan -->
<!-- ```{r} -->
<!-- decon_bspan = decon_results$BSpan_CIBERSORT %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSpan) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$BSpan_DRS %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSpan) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$BSpan_DTG %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$BSpan) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- - GTEx -->
<!-- ```{r} -->
<!-- decon_results$GTEx_CIBERSORT %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$GTEx) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$GTEx_DRS %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$GTEx) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$GTEx_DTG %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$GTEx) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- - HDBR -->
<!-- ```{r} -->
<!-- decon_results$HDBR_CIBERSORT %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$HDBR) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("")+ -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$HDBR_DRS %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$HDBR) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("")+ -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$HDBR_DTG %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$HDBR) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- - PsychEncode -->
<!-- ```{r} -->
<!-- decon_results$PE_CIBERSORT %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$PE) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$PE_DRS %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$PE) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- decon_results$PE_DTG %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("SampleID") %>% -->
<!--   dplyr::select(SampleID, Neurons, Astrocytes, -->
<!--                 Oligodendrocytes, Microglia, Endothelia) %>% -->
<!--   melt() %>% -->
<!--   left_join(md$PE) %>% -->
<!--   drop_na(Regions) %>% -->
<!--   mutate(variable= factor(variable, levels = c("Endothelia", "Microglia", -->
<!--                                               "Oligodendrocytes", -->
<!--                                               "Astrocytes", -->
<!--                                               "Neurons")), -->
<!--          Regions = factor(Regions, levels = c("Cortex", "Subcortex", -->
<!--                                               "Cerebellum", "Spinal Cord") )) %>% -->
<!--   ggplot(aes(x = variable, y = value, fill = variable)) + geom_boxplot() + -->
<!--   #scale_fill_manual(values = c("Cortex" = "#E85772", -->
<!--    #                                          "Subcortex" = "#878DC5", -->
<!--     #                                        "Cerebellum" = "#FFD357", -->
<!--      #                                        "Spinal Cord" = "#C6D14A")) + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(size = 10), -->
<!--         axis.text.y = element_text(size = 10), -->
<!--         legend.position = "none") + coord_flip() + -->
<!--   facet_grid(~Regions, scales = "free") +ylab("Estimated Proportion") + xlab("") + -->
<!--   scale_fill_manual(values = c("Neurons" = "#D94A70", -->
<!--                                "Astrocytes" = "#5590CC", -->
<!--                                "Oligodendrocytes" = "#AF9DCB", -->
<!--                                "Microglia" = "#69C9CF", -->
<!--                                "Endothelia" = "#FAE47B")) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
 [1] parallel  grid      stats4    tools     stats     graphics  grDevices
 [8] utils     datasets  methods   base     

other attached packages:
 [1] firatheme_0.2.4             Metrics_0.1.4              
 [3] naniar_1.0.0                glmpca_0.2.0               
 [5] broom_1.0.5                 cowplot_1.1.2              
 [7] glue_1.7.0                  ggfortify_0.4.16           
 [9] stargazer_5.2.3             AnnotationHub_3.10.0       
[11] BiocFileCache_2.10.1        dbplyr_2.4.0               
[13] singscore_1.22.0            EnsDb.Hsapiens.v86_2.99.0  
[15] ensembldb_2.26.0            AnnotationFilter_1.26.0    
[17] DT_0.31                     variancePartition_1.32.2   
[19] BiocParallel_1.36.0         gridExtra_2.3              
[21] ggvenn_0.1.10               recount3_1.12.0            
[23] e1071_1.7-14                DeconRNASeq_1.44.0         
[25] pcaMethods_1.94.0           limSolve_1.5.7             
[27] DTWBI_1.1                   ModelMetrics_1.2.2.2       
[29] preprocessCore_1.64.0       pander_0.6.5               
[31] dtangle_2.0.9               SummarizedExperiment_1.32.0
[33] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[35] GenomicFeatures_1.54.1      AnnotationDbi_1.64.1       
[37] Biobase_2.62.0              GenomicRanges_1.54.1       
[39] GenomeInfoDb_1.38.5         IRanges_2.36.0             
[41] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[43] biomaRt_2.58.0              plyr_1.8.9                 
[45] edgeR_4.0.11                limma_3.58.1               
[47] corrplot_0.92               lubridate_1.9.3            
[49] forcats_1.0.0               purrr_1.0.2                
[51] readr_2.1.5                 tidyverse_2.0.0            
[53] stringr_1.5.1               tidyr_1.3.0                
[55] scales_1.3.0                data.table_1.14.10         
[57] readxl_1.4.3                tibble_3.2.1               
[59] magrittr_2.0.3              reshape2_1.4.4             
[61] ggplot2_3.4.4               dplyr_1.1.4                
[63] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] fs_1.6.3                      ProtGenerics_1.34.0          
  [3] bitops_1.0-7                  httr_1.4.7                   
  [5] numDeriv_2016.8-1.1           backports_1.4.1              
  [7] utf8_1.2.4                    R6_2.5.1                     
  [9] lazyeval_0.2.2                withr_3.0.0                  
 [11] prettyunits_1.2.0             cli_3.6.2                    
 [13] labeling_0.4.3                entropy_1.3.1                
 [15] sass_0.4.8                    mvtnorm_1.2-4                
 [17] proxy_0.4-27                  dtw_1.23-1                   
 [19] Rsamtools_2.18.0              R.utils_2.12.3               
 [21] sessioninfo_1.2.2             rstudioapi_0.15.0            
 [23] RSQLite_2.3.5                 generics_0.1.3               
 [25] BiocIO_1.12.0                 gtools_3.9.5                 
 [27] Matrix_1.6-5                  fansi_1.0.6                  
 [29] abind_1.4-5                   R.methodsS3_1.8.2            
 [31] lifecycle_1.0.4               whisker_0.4.1                
 [33] yaml_2.3.8                    gplots_3.1.3                 
 [35] SparseArray_1.2.3             blob_1.2.4                   
 [37] promises_1.2.1                crayon_1.5.2                 
 [39] lattice_0.22-5                annotate_1.80.0              
 [41] KEGGREST_1.42.0               pillar_1.9.0                 
 [43] knitr_1.45                    rjson_0.2.21                 
 [45] boot_1.3-28.1                 corpcor_1.6.10               
 [47] lpSolve_5.6.20                codetools_0.2-19             
 [49] getPass_0.2-4                 vctrs_0.6.5                  
 [51] png_0.1-8                     Rdpack_2.6                   
 [53] cellranger_1.1.0              gtable_0.3.4                 
 [55] cachem_1.0.8                  xfun_0.41                    
 [57] mime_0.12                     rbibutils_2.2.16             
 [59] S4Arrays_1.2.0                iterators_1.0.14             
 [61] statmod_1.5.0                 ellipsis_0.3.2               
 [63] interactiveDisplayBase_1.40.0 lsa_0.73.3                   
 [65] nlme_3.1-164                  pbkrtest_0.5.2               
 [67] bit64_4.0.5                   progress_1.2.3               
 [69] EnvStats_2.8.1                filelock_1.0.3               
 [71] rprojroot_2.0.4               SnowballC_0.7.1              
 [73] bslib_0.6.1                   KernSmooth_2.23-22           
 [75] colorspace_2.1-0              DBI_1.2.1                    
 [77] tidyselect_1.2.0              processx_3.8.3               
 [79] extrafontdb_1.0               bit_4.0.5                    
 [81] compiler_4.3.3                curl_5.2.0                   
 [83] git2r_0.33.0                  graph_1.80.0                 
 [85] xml2_1.3.6                    DelayedArray_0.28.0          
 [87] rtracklayer_1.62.0            caTools_1.18.2               
 [89] remaCor_0.0.16                quadprog_1.5-8               
 [91] callr_3.7.3                   rappdirs_0.3.3               
 [93] digest_0.6.34                 minqa_1.2.6                  
 [95] rmarkdown_2.25                aod_1.3.3                    
 [97] XVector_0.42.0                RhpcBLASctl_0.23-42          
 [99] htmltools_0.5.7               pkgconfig_2.0.3              
[101] extrafont_0.19                lme4_1.1-35.1                
[103] highr_0.10                    fastmap_1.1.1                
[105] rlang_1.1.3                   htmlwidgets_1.6.4            
[107] shiny_1.8.0                   farver_2.1.1                 
[109] jquerylib_0.1.4               jsonlite_1.8.8               
[111] R.oo_1.25.0                   RCurl_1.98-1.14              
[113] rlist_0.4.6.2                 GenomeInfoDbData_1.2.11      
[115] munsell_0.5.0                 Rcpp_1.0.12                  
[117] visdat_0.6.0                  stringi_1.8.3                
[119] zlibbioc_1.48.0               MASS_7.3-60.0.1              
[121] Biostrings_2.70.1             splines_4.3.3                
[123] hms_1.1.3                     locfit_1.5-9.8               
[125] ps_1.7.6                      BiocVersion_3.18.1           
[127] XML_3.99-0.16                 evaluate_0.23                
[129] BiocManager_1.30.22           nloptr_2.0.3                 
[131] tzdb_0.4.0                    httpuv_1.6.13                
[133] Rttf2pt1_1.3.12               xtable_1.8-4                 
[135] restfulr_0.0.15               fANCOVA_0.6-1                
[137] later_1.3.2                   class_7.3-22                 
[139] lmerTest_3.1-3                memoise_2.0.1                
[141] GenomicAlignments_1.38.2      timechange_0.3.0             
[143] GSEABase_1.64.0               here_1.0.1                   </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
