---
title: "dataset-attributes"
author: "unawaz1996"
date: "2023-09-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Set-up

```{r}
###
libs = c("dplyr", "ggplot2", "reshape2", "tools", "magrittr", "tibble", "readxl", 
         "data.table", "scales", "tidyr", "reshape2", "stringr", "tidyverse", "readxl", "corrplot", "viridis", 
         "sunburstR", "pheatmap")
libsLoaded <- lapply(libs,function(l){suppressWarnings(suppressMessages(library(l, character.only = TRUE)))})

#source("code/preprocess/cerebroviz.R")
```

```{r}
summarise_stats = function(x, dataset)
{
    age = table(x$AgeInterval) %>% melt()
    age$Type = c(paste(dataset, "Sample", sep = "_"))
    individuals = x %>% group_by(AgeInterval, DonorID) %>% 
        dplyr::summarise(n = n()) %>% 
        as.data.frame() 
    individuals = table(individuals$AgeInterval) %>% melt()
    individuals$Type = c(paste("Individual", dataset, sep = "_"))
    
    age = rbind(age, individuals)
    colnames(age) = c("AgeInterval", "n", "Type")
    age$dataset = c(as.character(dataset))
    return(age)
    
}

summarise_regions = function(x, dataset){
    regions = x %>% 
        dplyr::group_by(AgeInterval, Regions) %>% 
        dplyr::summarise(n = n())  %>% 
        as.data.frame() %>% 
        mutate(Dataset = dataset)
    return(regions)
}
```


# Age intervals and number of samples and donors 


```{r}
age_interval_stats = list()
directory = file.path("/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/data/metadata")
pattern = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/data/metadata/"

for (f in directory){
    md = list.files(f, full.names = TRUE, pattern = "\\-metadata.csv$")
    
    for (j in md){
        ct_file = read.csv(j, header= TRUE)
        dataset = gsub(pattern, "", j)
        dataset = gsub("\\-metadata.csv","", dataset)
        message("Now calculating statistics for ", dataset)
        stats = summarise_stats(ct_file, dataset)
        age_interval_stats[[paste(dataset)]] = stats
    }
    
}


bulk_plot_version2 = age_interval_stats %>% 
    do.call(rbind, .) %>% 
    mutate(AgeInterval = factor(AgeInterval, levels = c("4-7pcw", "8-9pcw",
                                                        "10-12pcw", "13-15pcw", "16-18pcw",
                                                        "19-24pcw", "25-38pcw", "0-5mos",
                                                        "6-18mos", "19mos-5yrs", "6-11yrs",
                                                        "12-19yrs", "20-29yrs", "30-39yrs", "40-49yrs",
                                                        "50-59yrs", "60-69yrs", "70-79yrs", "80-89yrs", "90-99yrs")), 
           Type = factor(Type, levels = c("BrainSeq_Sample", "Individual_BrainSeq",
                                          "BrainSpan_Sample", "Individual_BrainSpan",
                                          "GTEx_Sample", "Individual_GTEx", 
                                          "HDBR_Sample", "Individual_HDBR", 
                                          "Ramakar_Sample", "Individual_Ramakar",
                                          "PsychEncode_Sample", "Individual_PsychEncode"))) %>%
    drop_na() %>%
    ggplot(aes(x= AgeInterval, y = n, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity") + 
    facet_grid(dataset~AgeInterval, scales = "free") + xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_rect(color = "#f7f4ed", fill = NA, size = 2)) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c("#F75E5E", "#FFC6BD",
                                 "#49165E", "#EBBAFF",
                                 "#78A2EB", "#36466A",
                                 "#74C69D", "#2D6A4F",
                                 "#F9AD79", "#FF5F0F")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA)
    )


bulk_plot_version2 

#ggsave(width = 11.01, height = 9.30, units = "in", 
 #      file = "../../Results/exploratory/bulk_dist_thesis.svg", plot = bulk_plot_version2)
#ggsave(file = "../../Results/exploratory/bulk_dist_update.svg", bulk_plot, 
 #      height = 18.9624, width = 45.3501, units = "cm")
    
```

```{r}
total_samples_bulk = age_interval_stats %>% 
    do.call(rbind, .) %>%
    drop_na() %>%
    group_by(Type) %>% 
    dplyr::summarise(sum = sum(n, na.rm = TRUE)) %>%
    as.data.frame()
```

- BrainSeq
```{r}
brainseq_total = total_samples_bulk %>%
    drop_na() %>%
    dplyr::filter(grepl("BrainSeq", Type)) %>% 
    mutate(Type = factor(Type, levels = c("Individual_BrainSeq","BrainSeq_Sample", 
                                    "Individual_BrainSpan", "BrainSpan_Sample",
                                    "Individual_GTEx", "GTEx_Sample",
                                    "Individual_PsychEncode", "PsychEncode_Sample"))) %>%
    ggplot(aes(x= Type, y = sum, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#FFC6BD","#F75E5E",
                                 "#49165E", "#EBBAFF",
                                 "#78A2EB", "#36466A",
                                 "#F9AD79", "#FF5F0F")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 2700) + coord_flip()

brainseq_total

ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/BrainSeq_total.svg"), brainseq_total, width = 1.98, 
       height = 1.89,, units = "in")

```

- BrainSpan


```{r}
brainSpan_total = total_samples_bulk %>%
    drop_na() %>%
    dplyr::filter(grepl("BrainSpan", Type)) %>% 
    mutate(Type = factor(Type, levels = c("Individual_BrainSeq","BrainSeq_Sample", 
                                          "Individual_BrainSpan", "BrainSpan_Sample",
                                          "Individual_GTEx", "GTEx_Sample",
                                          "Individual_PsychEncode", "PsychEncode_Sample"))) %>%
    ggplot(aes(x= Type, y = sum, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c("#EBBAFF", "#49165E")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 2700) + coord_flip()

brainSpan_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/BrainSpan_total.svg"), brainSpan_total, width = 1.98, 
       height = 1.89, units = "in")

```

- GTEx


```{r}
gtex_total = total_samples_bulk %>%
    drop_na() %>%
    dplyr::filter(grepl("GTEx", Type)) %>% 
    mutate(Type = factor(Type, levels = c("Individual_BrainSeq","BrainSeq_Sample", 
                                          "Individual_BrainSpan", "BrainSpan_Sample",
                                          "Individual_GTEx", "GTEx_Sample",
                                          "Individual_PsychEncode", "PsychEncode_Sample"))) %>%
    ggplot(aes(x= Type, y = sum, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#36466A", "#78A2EB")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 2700) + coord_flip()


ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/GTEx_total.svg"), gtex_total, width = 1.98, 
       height = 1.89, units = "in")


```

- HDBR

```{r}
HDBR_total = total_samples_bulk %>%
    drop_na() %>%
    dplyr::filter(grepl("HDBR", Type)) %>% 
    mutate(Type = factor(Type, levels = c("Individual_BrainSeq","BrainSeq_Sample", 
                                          "Individual_BrainSpan", "BrainSpan_Sample",
                                          "Individual_GTEx", "GTEx_Sample",
                                          "Individual_HDBR", "HDBR_Sample", 
                                          "Individual_PsychEncode", "PsychEncode_Sample"))) %>%
    ggplot(aes(x= Type, y = sum, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#F3765C", "#E11F28")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 2700) + coord_flip()
HDBR_total

ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/HDBR_total.svg"), HDBR_total , width = 1.98, 
       height = 1.89, units = "in")

```

- PsychEncode

```{r}
pe_total = total_samples_bulk %>%
    drop_na() %>%
    dplyr::filter(grepl("Psych", Type)) %>% 
    mutate(Type = factor(Type, levels = c("Individual_BrainSeq","BrainSeq_Sample", 
                                          "Individual_BrainSpan", "BrainSpan_Sample",
                                          "Individual_GTEx", "GTEx_Sample",
                                          "Individual_PsychEncode", "PsychEncode_Sample"))) %>%
    ggplot(aes(x= Type, y = sum, fill =Type)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#FF5F0F",  "#F9AD79")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 2700) + coord_flip()


pe_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/PE_total.svg"),pe_total, width = 1.98, 
       height = 1.89, units = "in")
```

## Single-nucleus RNA-seq


```{r}
snRNAseq = list()

snRNAseq$Vel = read.csv(here::here("data/metadata/snRNA-seq/Velmeshev-metadata.csv"), 
                        header = TRUE)
snRNAseq$HCA = read.csv(here::here("data/metadata/snRNA-seq/HCA-metadata.csv"), 
                        header = TRUE)
snRNAseq$Cameron = read.csv(here::here("data/metadata/snRNA-seq/Cameron-metadata.csv"), 
                        header = TRUE)
snRNAseq$Aldringer = read.csv(here::here("data/metadata/snRNA-seq/Aldringer-metadata.csv"), 
                        header = TRUE)

```


```{r}
vel_total= data.frame(dataset = c("Vel", "HCA", "Cameron", "Aldringer"), 
           val = c(nrow(snRNAseq$Vel), nrow(snRNAseq$HCA),
                   nrow(snRNAseq$Cameron),nrow(snRNAseq$Aldringer) )) %>%
    dplyr::filter(grepl("Vel", dataset)) %>% 
    ggplot(aes(x= dataset, y = val, fill =dataset)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#76C7A1")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 103000) + coord_flip()

vel_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/Vel_total.svg"),vel_total, width = 1.98, 
       height = 1.89, units = "in")

```

```{r}
hca_total= data.frame(dataset = c("Vel", "HCA", "Cameron", "Aldringer"), 
           val = c(nrow(snRNAseq$Vel), nrow(snRNAseq$HCA),
                   nrow(snRNAseq$Cameron),nrow(snRNAseq$Aldringer) )) %>%
    dplyr::filter(grepl("HCA", dataset)) %>% 
    ggplot(aes(x= dataset, y = val, fill =dataset)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#FFD357")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 103000) + coord_flip()

hca_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/HCA_total.svg"),hca_total, width = 1.98, 
       height = 1.89, units = "in")

```

```{r}
cam_total= data.frame(dataset = c("Vel", "HCA", "Cameron", "Aldringer"), 
           val = c(nrow(snRNAseq$Vel), nrow(snRNAseq$HCA),
                   nrow(snRNAseq$Cameron),nrow(snRNAseq$Aldringer) )) %>%
    dplyr::filter(grepl("Cameron", dataset)) %>% 
    ggplot(aes(x= dataset, y = val, fill =dataset)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#E85772")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 103000) + coord_flip()

cam_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/Cameron_total.svg"),cam_total, width = 1.98, 
       height = 1.89, units = "in")

```

```{r}
aldringer_total= data.frame(dataset = c("Vel", "HCA", "Cameron", "Aldringer"), 
           val = c(nrow(snRNAseq$Vel), nrow(snRNAseq$HCA),
                   nrow(snRNAseq$Cameron),nrow(snRNAseq$Aldringer) )) %>%
    dplyr::filter(grepl("Aldringer", dataset)) %>% 
    ggplot(aes(x= dataset, y = val, fill =dataset)) +
    geom_bar(position = "dodge",stat= "identity", width = .5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", 
          #axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
    scale_fill_manual(values = c( "#878DC5")) + 
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank()) + 
       # axis.text.x = element_blank(), 
        #axis.text.y = element_blank()) + 
       ylim(0, 103000) + coord_flip()

aldringer_total
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/Aldringer_total.svg"),aldringer_total, width = 1.98, 
       height = 1.89, units = "in")


```

```{r}

brain.dir = file.path("/home/neuro/Documents/BrainData/Bulk")
md = list()
md$BrainSeq = read.csv(file.path(brain.dir, "Brainseq/Formatted/BrainSeq-metadata.csv"),
                       header=TRUE)
md$BrainSpan = read.csv(file.path(brain.dir, "BrainSpan/Formatted/BrainSpan-metadata.csv"),
                        header=TRUE)
md$GTEx = read.csv(file.path(brain.dir, "GTEx/Formatted/GTEx-metadata.csv"), 
                   header = TRUE)
md$HDBR = read.csv(file.path(brain.dir, "HDBR/Formatted/HDBR-metadata.csv"), 
                   header = TRUE)
md$PE = read.csv(file.path(brain.dir, "PsychEncode/Formatted/PsychEncode-metadata.csv"), 
                   header = TRUE)
```


<!-- ```{r} -->

<!-- bseq =table(md$BrainSeq$StructureAcronym) %>%  -->
<!--   as.data.frame() %>%  -->
<!--   set_colnames(c("Structure", "Value")) %>%  -->
<!--   mutate(Structure = gsub("DLPFC", "DFC", Structure)) %>% -->
<!--   column_to_rownames("Structure")  -->

<!-- cerebro_Viz(as.matrix(bseq)) -->
<!-- ``` -->


<!-- ```{r} -->
<!-- bseq =table(md$BrainSpan$StructureAcronym) %>%  -->
<!--   as.data.frame() %>%  -->
<!--   set_colnames(c("Structure", "Value")) %>%  -->
<!--   mutate(Structure = gsub("DLPFC", "DFC", Structure)) %>% -->
<!--   column_to_rownames("Structure")  -->

<!-- cerebro_Viz(as.matrix(bseq)) -->

<!-- ``` -->


## Regions per age interval 

```{r}
regions_per_interval = list()

for (f in directory){
    md = list.files(f, full.names = TRUE, pattern = "\\-metadata.csv$")
    
    for (j in md){
        ct_file = read.csv(j, header= TRUE)
        dataset = gsub(pattern, "", j)
        dataset = gsub("\\-metadata.csv","", dataset)
        message("Now calculating statistics for ", dataset)
        stats = summarise_regions(ct_file, dataset)
        regions_per_interval[[paste(dataset)]] = stats
    }
    
}
```


```{r}
regions_count = regions_per_interval %>% 
    do.call(rbind,.) %>% 
    mutate(AgeInterval = factor(AgeInterval, 
                                levels = c("4-7pcw", "8-9pcw", 
                                           "10-12pcw", "13-15pcw", "16-18pcw",
                                           "19-24pcw", "25-38pcw", "0-5mos",
                                           "6-18mos", "19mos-5yrs", "6-11yrs",
                                           "12-19yrs", "20-29yrs", 
                                           "30-39yrs", "40-49yrs",
                                           "50-59yrs", "60-69yrs", 
                                           "70-79yrs", "80-89yrs", "90-99yrs"))) %>%
    mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Chroid plexus", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain"
                            ))) %>%
    
    drop_na() %>%
    ggplot(aes(y=n, x=AgeInterval, fill = Regions)) +
    geom_bar(position = "dodge",stat= "identity") + 
    #geom_col(position = "fill") +
       scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))+
    facet_grid(Dataset~AgeInterval, scales = "free") + xlab("") + ylab("")  + theme_bw() +
    theme(axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_rect(color = "#f7f4ed", fill = NA, size = 2),
          legend.position = "top") + 
    theme(strip.text = element_text(colour = 'white')) +
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA)
    ) 


ggsave(width = 11.01, height = 10.30, units = "in", 
       file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/bulk_regions_dataset_count.svg"), 
       plot = regions_count )
```


```{r}
total_regions =regions_per_interval  %>% 
    do.call(rbind, .) %>%
    drop_na() %>%
    group_by(Regions, Dataset) %>% 
    dplyr::summarise(sum = sum(n, na.rm = TRUE)) %>%
    as.data.frame()

max(total_regions$sum)
```


```{r}
bseq_total =total_regions %>%
    drop_na() %>%
    dplyr::filter(grepl("BrainSeq", Dataset)) %>% 
  mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Forebrain and midbrain", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain fragment"
                            )))  %>%
  ggplot(aes(x= Dataset, y = sum, fill =Regions)) +
    geom_bar(position = "dodge",stat= "identity", color = "white") +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
      scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                ))+
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 1400) + coord_flip()


ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/BrainSeq_regions_total.svg"),bseq_total, width = 1.98, 
       height = 1.89, units = "in")
```





```{r}
bspan_total = total_regions %>%
    drop_na() %>%
    dplyr::filter(grepl("BrainSpan", Dataset)) %>% 
  mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Forebrain and midbrain", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain fragment"
                            )))  %>%
  ggplot(aes(x= Dataset, y = sum, fill =Regions)) +
    geom_bar(position = "dodge",stat= "identity", color = "white") +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
      scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                ))+
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 1400) + coord_flip()

ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/BrainSpan_regions_total.svg"),bspan_total, width = 1.98, 
       height = 1.89, units = "in")
```





```{r}
gtex_total =total_regions %>%
    drop_na() %>%
    dplyr::filter(grepl("GTEx", Dataset)) %>% 
  mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Chroid plexus", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain fragment"
                            )))  %>%
  ggplot(aes(x= Dataset, y = sum, fill =Regions)) +
    geom_bar(position = "dodge",stat= "identity", color = "white") +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
      scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))+
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 1400) + coord_flip()

ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/GTEx_regions_total.svg"),gtex_total, width = 1.98, 
       height = 1.89, units = "in")

```

```{r}
hdbr_total = total_regions %>%
    drop_na() %>%
    dplyr::filter(grepl("HDBR", Dataset)) %>% 
  mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Chroid plexus", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain"
                            )))  %>%
  ggplot(aes(x= Dataset, y = sum, fill =Regions)) +
    geom_bar(position = "dodge",stat= "identity", color = "white") +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
      scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 1400) + coord_flip()

hdbr_total 
ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/HDBR_regions_total.svg"),hdbr_total, width = 1.98, 
       height = 1.89, units = "in")
```


```{r}
pe_total = total_regions %>%
    drop_na() %>%
    dplyr::filter(grepl("PsychEncode", Dataset)) %>% 
  mutate(Regions = factor(Regions, 
                            levels= c("Cortex", 
                                      "Subcortex",
                                      "Cerebellum", 
                                      "Spinal Cord", 
                                      "Forebrain", 
                                      "Forebrain and midbrain", 
                                      "Midbrain",
                                      "Hindbrain", 
                                      "Brain fragment"
                            )))  %>%
  ggplot(aes(x= Dataset, y = sum, fill =Regions)) +
    geom_bar(position = "dodge",stat= "identity", color = "white", width = 0.5) +  xlab("") + ylab("")  + theme_bw() +
    theme(legend.position = "none", #axis.text.x=element_blank(),
          strip.background =element_rect(fill="#AA9A9C", color = "#E1DFDB"), 
          panel.border = element_blank()) + 
    theme(strip.text = element_text(colour = 'white')) +
      scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
        plot.background = element_rect(fill = "transparent",colour =NA), 
        panel.grid.minor = element_blank(), 
      #  axis.text.x = element_blank(), 
        axis.text.y = element_blank()) + ylim(0, 1400) + coord_flip()

ggsave(file = here::here("output/Thesis_plots/Attributes/Donor_and_sample/PsychEncode_regions_total.svg"),pe_total, width = 1.98, 
       height = 1.89, units = "in")

```


```{r}
regions_per_interval %>% 
    do.call(rbind,.) %>% 
        dplyr::group_by(Dataset, Regions) %>% 
        dplyr::summarise(n = n())  %>% 
        as.data.frame() %>% 
        mutate(Dataset = dataset)
```


```{r}
regions_per_dataset = list()

for (f in directory){
    md = list.files(f, full.names = TRUE, pattern = "\\-metadata.csv$")
    
    for (j in md){
        ct_file = read.csv(j, header= TRUE)
        dataset = gsub(pattern, "", j)
        dataset = gsub("\\-metadata.csv","", dataset)
        message("Now calculating statistics for ", dataset)
        stats = ct_file %>% 
        dplyr::group_by(Regions) %>% 
        dplyr::summarise(n = n()) %>% 
          as.data.frame() %>% 
        mutate(Dataset = dataset)
        
        regions_per_dataset[[paste(dataset)]] = stats
    }
    
}

regions_per_dataset %<>% 
  do.call(rbind, .) %>% 
  rownames_to_column("names") %>% 
  dplyr::select(-c("names"))


write.csv(regions_per_dataset, "total_regions.csv")


md_cam = read.csv("/home/neuro/Documents/BrainData/single-cell/cameron/Cameron-metadata.csv", header=TRUE)
```
