---
title: "gene-exp"
author: "urwahnawaz"
date: "2023-10-03"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Document overview 

This notebook contains plotting of the case studies for BITHub. 

```{r libraries include=FALSE}
libs = c("dplyr", "ggplot2", "reshape2", "tools", "magrittr", "tibble", "readxl", 
         "data.table", "scales", "tidyr", "reshape2", "stringr", "tidyverse", "readxl", "corrplot", "viridis", 
         "sunburstR", "pheatmap", "ggpubr")
libsLoaded <- lapply(libs,function(l){suppressWarnings(suppressMessages(library(l, character.only = TRUE)))})
library(EnsDb.Hsapiens.v86)
source(here::here("code/preprocess/functions.R"))
source(here::here("code/preprocess/def_stages.R"))
```

```{r annotations}
gene2symbol = genes(EnsDb.Hsapiens.v86, return.type="DataFrame") %>%
    as.data.frame()
gene2symbol %<>% 
    dplyr::select("EnsemblID" = "gene_id", gene_name)

```

```{r load-exp}
brain.dir =file.path("/home/neuro/Documents/BrainData/Bulk")
e <- list()
  
e$Raw$GTEx <- read.csv(file.path(brain.dir, "/GTEx/Formatted/GTEx-exp.csv"), 
                       check.names = FALSE, row.names = 1) 
e$Raw$PE <- read.csv(file.path(brain.dir, "/PsychEncode/Formatted/PsychEncode-exp.csv"), row.names = 1,
                 check.names = FALSE) 
e$Raw$BSpan <- read.csv(file.path(brain.dir, "/BrainSpan/Formatted/BrainSpan-exp.csv"),
                        check.names = FALSE, row.names=1)
e$Raw$BSeq <- read.csv(file.path(brain.dir,"/Brainseq/Formatted/BrainSeq-exp.csv"), 
                       check.names = FALSE, row.names = 1)

e$Raw$HDBR <- read.csv(file.path(brain.dir, "/HDBR/Formatted/HDBR-exp.csv"), 
                       check.names = FALSE, row.names = 1) %>% 
  column_to_rownames("EnsemblID")


```

```{r load-md}
md = list()
md$GTEx <- read.csv(file.path(brain.dir, "/GTEx/Formatted/GTEx-metadata.csv"), 
                       check.names = FALSE, row.names = 1)
md$PE <- read.csv(file.path(brain.dir,"/PsychEncode/Formatted/PsychEncode-metadata.csv"),
                  check.names = FALSE)
md$BSeq <- read.csv(file.path(brain.dir,"/Brainseq/Formatted/BrainSeq-metadata.csv"), 
                       check.names = FALSE)
md$HDBR <- read.csv(file.path(brain.dir, "/HDBR/Formatted/HDBR-metadata.csv"), 
                       check.names = FALSE, row.names = 1) 
md$Ram <- read.csv(file.path(brain.dir, "/Ramakar/Formatted/Ramaker-metadata.csv"),
                       check.names = FALSE, row.names = 1)

md$BSpan <- read.csv(file.path(brain.dir,
                               "/BrainSpan/Formatted/BrainSpan-metadata.csv"),
                     check.names = FALSE)
```


```{r}
e$Scale <- lapply(e$Raw, scale) # basic z-score columnwise transformation
e$Log <- lapply(e$Raw, function(x) { log2(x+0.5) }) # log2
e$Log_Scale <- lapply(e$Log, scale) # log2 followed by scale
```



```{r include=FALSE}
zscore_overall = list()

zscore_overall = lapply(e$Log_Scale, function(x){
  name = names(x)
  rowMeans(x) %>% 
    melt()
})
```


```{r include=FALSE}
regions_scale = list()
datasets = unique(names(e$Raw))

for (data in datasets){
  name = data
  mat = e$Raw[[data]]
  md_sub = md[[data]]
  md_sub %<>% drop_na(Regions)
  regions = unique(md_sub$Regions)
  
  for(reg in regions){
    md_region = md_sub[md_sub$Regions == reg,]
    mat_sub = mat %>% 
      dplyr::select(contains(md_region$SampleID))
    mat_sub =  log2(mat_sub+0.5)
    mat_sub = scale(mat_sub)
    mat_sub = mat_sub %>% rowMeans() %>% melt()
    regions_scale[[paste0(name,"_",reg)]] = mat_sub
    
    
  }
}
```


```{r include=FALSE}
period_region_scale = list()
datasets = unique(names(e$Raw))

for (data in datasets){
  name = data
  mat = e$Raw[[data]]
  md_sub = md[[data]]
  md_sub %<>% drop_na(Period)
  md_sub %<>% drop_na(Regions)
  
  for(i in unique(md_sub$Period)){
    for(r in unique(md_sub$Regions)){
      md_rP = md_sub[md_sub$Period == i & md_sub$Regions == r,]
      mat_sub = mat %>% 
      dplyr::select(contains(md_rP$SampleID))
      mat_sub =  log2(mat_sub+0.5)
      mat_sub = scale(mat_sub)
      mat_sub = mat_sub %>% rowMeans() %>% melt()
      period_region_scale[[paste0(name,"_",i, "_", r)]] = mat_sub
      
    }
  }
}
```

```{r include=FALSE}
period_scale = list()
datasets = unique(names(e$Raw))

for (data in datasets){
  name = data
  mat = e$Raw[[data]]
  md_sub = md[[data]]
  md_sub %<>% drop_na(Period)
  
  for(i in unique(md_sub$Period)){
    md_p = md_sub[md_sub$Period == i,]
    mat_sub = mat %>% 
      dplyr::select(contains(md_p$SampleID))
    mat_sub =  log2(mat_sub+0.5)
    mat_sub = scale(mat_sub)
    mat_sub = mat_sub %>% rowMeans() %>% melt()
    period_scale[[paste0(name,"_",i)]] = mat_sub
    
    
  }
}

```

```{r}
genes_of_interest = c("FMR1", "ZMYND8", "UPF1", "FOXG1", "ZIC2", "MKI67", "KIF1B")
exp_data = list()
datasets = c("BSpan", "GTEx", "BSeq", "PE", "HDBR")
for (x in datasets) {
  message(paste("Subsetting samples for", x))
  dataset = e$Raw[[x]]
  md_sub = md[[x]]
  name = x
  sub = dataset %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% genes_of_interest) %>% 
    melt() %>%
    dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    mutate(Dataset = x) %>% 
    left_join(md_sub %>% 
                dplyr::select(SampleID, Regions,
                              StructureAcronym, Period, 
                              AgeInterval), by = "SampleID")
  exp_data[[paste0(name)]] = sub 
  
}
```



## Region and period specific brain development

To demonstrate the utility of BITHub, I utilised ZIC2 and MKI67 genes as examples. 

- ZIC2 
```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + ylim(-1, 1.5) +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))




```


```{r}
bspan_cer =  exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "GTEx")) %>%
  dplyr::filter(Regions == "Cerebellum") %>% 
  mutate(logExp = log10(Exp))
  median(bspan_cer$logExp)
```

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```



```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(TPM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))


# ggsave(gtex, file=here::here("output/Thesis_plots/Examples/GTEx-ZIC2.svg"), 
#         width = 8, 
#        height =5, 
#        units = "in")
```



```{r}
hdbr  = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + ylim(-1, 3) +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


# ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/HDBR-ZIC2.svg"), 
#         width = 8, 
#        height =5, 
#        units = "in")
```


```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
    theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


# ggsave(pe, file=here::here("output/Thesis_plots/Examples/PsychEncode-ZIC2.svg"), 
#         width = 8, 
#        height =5, 
#        units = "in")

```

```{r fig-height=15, fig.width=12}
#ggsave(filename = here::here("output/Thesis_plots/Examples/ZIC2-all.svg"),
 #      height = 14, width = 16, units = "in")
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

ggsave(filename = here::here("output/Thesis_plots/Examples/ZIC2-all.png"),
       height = 14, width = 16, units = "in", dpi=1200)
#dev.off()
```

```{r}
dataset_zscore_zic2 = zscore_overall$GTEx %>% 
  set_colnames("z.score.gtex") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(zscore_overall$BSpan %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000043355", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.bspan,
                            y = z.score.gtex, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("ZIC2 (ENSG00000043355)") +
  labs(x = "BrainSpan", 
       y = "GTEx") 

```

```{r}
zscore_overall$GTEx %>% 
  set_colnames("z.score.gtex") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(zscore_overall$BSpan %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  dplyr::filter(EnsID == "ENSG00000043355" )
  

```

```{r}
regions_zscore_zic2 = regions_scale$BSeq_Subcortex %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(regions_scale$BSpan_Subcortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000043355", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.bspan,
                            y = z.score.bseq, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("ZIC2 (ENSG00000043355)") +
  labs(x = "BrainSpan (Subcortex)", 
       y = "BrainSeq (Subcortex)") 
```

```{r}
regions_scale$BSeq_Subcortex %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(regions_scale$BSpan_Subcortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  dplyr::filter(EnsID == "ENSG00000043355" )

```

```{r}
period_zscore_zic2 = period_scale$BSpan_Prenatal %>% 
  set_colnames("z.score.prenatal") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_scale$BSpan_Postnatal %>% 
                set_colnames("z.score.postnatal") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000043355", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.prenatal,
                            y = z.score.postnatal, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("ZIC2 (ENSG00000043355)") +
  labs(x = "BrainSpan (Prenatal)", 
       y = "BrainSeq (Postnatal)") 
```

```{r}
period_scale$BSpan_Prenatal %>% 
  set_colnames("z.score.prenatal") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_scale$BSpan_Postnatal %>% 
                set_colnames("z.score.postnatal") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  dplyr::filter(EnsID == "ENSG00000043355" )
```

```{r}
# png(file = here::here("output/Thesis_plots/Examples/ZIC2-zscore.png"),
#        height = 6, width = 18)
ggarrange(dataset_zscore_zic2 , regions_zscore_zic2,
          period_zscore_zic2, ncol=3, nrow=1 )
ggsave(filename = here::here("output/Thesis_plots/Examples/ZIC2-zscore.png"),
       height = 6, width = 18, dpi=1200)

```


- MKI67

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
bseq
# ggsave(bseq, file=here::here("output/Thesis_plots/Examples/BrainSeq-MKI67.svg"), 
#         width = 8, 
#        height =5, 
#        units = "in")

```

- BrainSpan
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bspan
# ggsave(bspan, file=here::here("output/Thesis_plots/Examples/BrainSpan-MKI67.svg"), 
#         width = 8, 
#        height =5, 
#        units = "in")
```

- GTEx

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
   geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + ylim(-3,2) +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

gtex
```

- HDBR 

```{r}
hdbr  = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + ylim(-2, 2) +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr
```

- PsychEncode 

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

pe

```

```{r}

ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

ggsave(filename = here::here("output/Thesis_plots/Examples/MKI67-all.png"),
       height = 14, width = 16, dpi=1200)
```

Zscore of MKI69
```{r}
pe_zscore_mki67 = period_scale$PE_Postnatal %>% 
  set_colnames("z.score.pe.postnatal") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_scale$PE_Prenatal %>% 
                set_colnames("z.score.pe.prenatal") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000148773", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.pe.prenatal,
                            y = z.score.pe.postnatal, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("MKI67 (ENSG00000148773)") +
  labs(x = "PsychENCODE (Prenatal)", 
       y = "PsychENCODE (Postnatal)") 
```

```{r}
 period_scale$PE_Postnatal %>% 
  set_colnames("z.score.postnatal") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_scale$PE_Prenatal %>% 
                set_colnames("z.score.prenatal") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  dplyr::filter(EnsID == "ENSG00000148773" )
```


```{r}
cortex_zscore_mki67 = period_region_scale$BSeq_Prenatal_Cortex %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_region_scale$BSpan_Prenatal_Cortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000148773", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.bseq,
                            y = z.score.bspan, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("MKI67 (ENSG00000148773)") +
  labs(x = "BrainSeq (Prenatal Cortex)", 
       y = "BrainSpan (Prenatal Cortex)") 
```

```{r}
period_region_scale$BSeq_Prenatal_Cortex %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_region_scale$BSpan_Prenatal_Cortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>%
  dplyr::filter(EnsID == "ENSG00000148773" )
```


```{r}
bspan_zscore_mki67 = period_region_scale$BSpan_Prenatal_Cerebellum %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_region_scale$BSpan_Prenatal_Subcortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>% 
  mutate(Color = ifelse(EnsID == "ENSG00000148773", "No", "Color")) %>% 
  arrange(Color) %>%
  #mutate(Color = factor(Color, levels = c( "Color", "No"))) %>%
  ggplot() + geom_point(aes(x= z.score.bseq,
                            y = z.score.bspan, color = Color, 
                            size = Color, alpha = Color)) +
  theme_bw() + scale_size_manual(values = c("No" = 5, 
                                      "Color" = 2)) + 
  scale_color_manual(values = c("Color" = "#F68D6F", 
                                "No" = "#6370E0")) + 
  scale_alpha_manual(values =c("No" = 1, 
                               "Color" = 0.45)) +
  theme(legend.position = "none", 
        axis.text = element_text(size =12, family  = "serif", colour = "black"),
        axis.title.x =  element_text(size =14, family  = "serif"),
        axis.title.y =  element_text(size =14, family  = "serif"),
        axis.ticks = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.major = element_line(colour = "grey70", linewidth = 0.2),
        panel.grid.minor = element_blank(), 
        axis.title = element_text(size = 13),
        plot.title = element_text(family = "serif", size =18)) +
  ggtitle("MKI67 (ENSG00000148773)") +
  labs(x = "BrainSpan (Prenatal Cerebellum)", 
       y = "BrainSpan (Prenatal Subcortex)") 
```

```{r}
period_region_scale$BSpan_Prenatal_Cerebellum %>% 
  set_colnames("z.score.bseq") %>% 
  rownames_to_column("EnsID") %>% 
  left_join(period_region_scale$BSpan_Prenatal_Subcortex %>% 
                set_colnames("z.score.bspan") %>% 
              rownames_to_column("EnsID"),by = "EnsID") %>%
  dplyr::filter(EnsID == "ENSG00000148773" )
```

```{r}

ggarrange(pe_zscore_mki67 , cortex_zscore_mki67,
          bspan_zscore_mki67, ncol=3, nrow=1 )
ggsave(file = here::here("output/Thesis_plots/Examples/MKI67-zscore.png"),
       height = 6, width = 18)

```

## RNA degradation in the brain 
We want to investigate the impact of genes having previously been altered with respect to PMI levels. 
Other than BrainSeq and PsychEncode, all other datasets contain PMI values. 
For ease, made the scale of each PMI consistent into minutes and have it as continuous and categorical values. 
Categorical values have the PMI binned into intervals of based on the GTEx Biospecimen Methodological Study (BMS). 

- Study by Fereria aet al showed that gene changes with PMI are tissue related
* Genes to test are RNASE2, CXCL2, EGR3 and HBA1 had no significant associated in their brain datasets tested 


```{r}
var_part = list()

var_part$BSpan = read.csv(file.path(brain.dir,"/BrainSpan/Formatted/BrainSpan-varPart.csv"), header=TRUE, 
                                    row.names =1)  %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID")
var_part$BSeq = read.csv(file.path(brain.dir,"/Brainseq/Formatted/BrainSeq-varPart.csv"), 
                                   header=TRUE, row.names=1) %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID")
var_part$GTEx = read.csv(file.path(brain.dir,"/GTEx/Formatted/GTEx-varPart.csv"), header=TRUE, 
                                   row.names=1) %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID")
var_part$HDBR = read.csv(file.path(brain.dir,"/HDBR/Formatted/HDBR-varPart.csv"), header=TRUE, 
                                   row.names=1) %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID")
var_part$PE = read.csv(file.path(brain.dir,"/PsychEncode/Formatted/PsychEncode-varPart.csv"), header=TRUE, 
                                   row.names=1) %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID")
```


-  RNASE2

```{r}
md$GTEx %<>% 
  mutate(PMI.hours = PMI/60) %>% 
  mutate(PMI.intervals = case_when(
    PMI.hours < 1 ~ "< 1", 
    PMI.hours >= 1 & PMI.hours< 4 ~ "\u2265 1 & < 4", 
    PMI.hours >= 4 & PMI.hours < 8 ~ "\u2265 4 & < 8", 
    PMI.hours >= 8 & PMI.hours < 12 ~ "\u2265 8 & < 12", 
    PMI.hours >= 12 & PMI.hours < 16 ~ "\u2265 12 & < 16", 
    PMI.hours >= 16 & PMI.hours < 20 ~ "\u2265 16 & < 20", 
    PMI.hours >= 20 ~ "\u2265 20"
  ))


gtex_fig_RNASE2  = e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "RNASE2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  ))) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("RNASE2 - GTEx") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") +
  ylab(bquote(log["10"](TPM)))


gtex_fig_RNASE2


cor_RNASE2_gtex = e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "RNASE2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  ))) %>% 
  mutate(logExp = log10(Exp)) %>%
  ggscatter(., x = "PMI.hours", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("RNASE2 - GTEx") +
  ylab("TPM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")


  
cor_RNASE2_gtex


    
```



```{r}
md$BSpan = md$BSpan[,-1]
md$BSpan  %<>%
#  drop_na(PMI) %>%
  mutate(PMI.hours = PMI/60) %>% 
  mutate(PMI.intervals = case_when(
    PMI < 1 ~ "< 1", 
    PMI >= 1 & PMI< 4 ~ "\u2265 1 & < 4", 
    PMI >= 4 & PMI < 8 ~ "\u2265 4 & < 8", 
    PMI >= 8 & PMI < 12 ~ "\u2265 8 & < 12", 
    PMI >= 12 & PMI < 16 ~ "\u2265 12 & < 16", 
    PMI >= 16 & PMI< 20 ~ "\u2265 16 & < 20", 
    PMI >= 20 ~ "\u2265 20"
  ))



bspan_fig_RNASE2 = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "RNASE2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>% 
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("RNASE2 - BrainSpan") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") +
  ylab(bquote(log["10"](RPKM)))



bspan_fig_RNASE2


cor_RNASE2_bspan = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "RNASE2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggscatter(., x = "PMI", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("RNASE2 - BrainSpan") +
  ylab("RPKM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")

```


- CXCL2

```{r}
gtex_plot =e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "CXCL2") %>% 
    melt() %>% 
   dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("CXCL2 - GTEx") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](TPM)))

gtex_plot


cor_CXCL2_gtex = e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "CXCL2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  ))) %>% 
  mutate(logExp = log10(Exp)) %>%
  ggscatter(., x = "PMI.hours", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("CXCL2 - GTEx") +
  ylab("TPM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")


cor_CXCL2_gtex 
```

```{r}
 sc.cxcl2 = e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "EGR3") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  dplyr::filter(Regions == "Spinal Cord") %>% 
  mutate(logExp = log10(Exp))

median(sc.cxcl2$logExp)

```

```{r}
bspan_fig =e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "CXCL2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
      mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("CXCL2 - BrainSpan") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](RPKM)))
bspan_fig  



cor_CXCL2_bspan = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "CXCL2") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggscatter(., x = "PMI", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("CXCL2 - BrainSpan") +
  ylab("RPKM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")

```

```{r}
gtex_var_cxcl2 = var_part$GTEx %>% 
  dplyr::filter(gene_name %in% "CXCL2") %>% 
  melt() %>%
  mutate(percentage = value * 100) %>%
  ggplot(aes(x= fct_reorder(variable,percentage  ), y = percentage )) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + xlab("") + ggtitle("CXCL2 - GTEx") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) 
```

```{r}
bspan_var_cxlc2 = var_part$BSpan %>% 
  dplyr::filter(gene_name %in% "CXCL2") %>% 
  melt() %>%
    ggplot(aes(x= fct_reorder(variable, value), y = value)) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + xlab("")  + ggtitle("CXCL2 - BrainSpan") +  
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ))
```


- EGR3


```{r}
gtex_plot_egr3 = e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "EGR3") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("EGR3 - GTEx") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](TPM)))

gtex_plot_egr3


cor_egr3_gtex =  e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "EGR3") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  ))) %>% 
  mutate(logExp = log10(Exp)) %>%
  ggscatter(., x = "PMI.hours", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("EGR3 - GTEx") +
  ylab("TPM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")
```

```{r}
bspan_plot_EGR3 = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "EGR3") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("EGR3 - BrainSpan") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](RPKM)))

bspan_plot_EGR3

cor_EGR3_bspan = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "EGR3") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
    ggscatter(., x = "PMI", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("EGR3 - BrainSpan") +
  ylab("RPKM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")


```

```{r}
gtex_var_egr3 =var_part$GTEx %>% 
  dplyr::filter(gene_name %in% "EGR3") %>% 
  melt() %>%
  ggplot(aes(x= fct_reorder(variable, value), y = value)) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + ggtitle("EGR3 - GTEx") + xlab("") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) 

```

```{r}
bspan_var_egr3 =var_part$BSpan %>% 
  dplyr::filter(gene_name %in% "EGR3") %>% 
  melt() %>% mutate(percent = value * 100)
  ggplot(aes(x= fct_reorder(variable, value), y = value)) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + xlab("") + ggtitle("EGR3 - BrainSpan") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) 

```

 - HBA1

```{r}
gtex_plot_HBA1 =e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "HBA1") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
 mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("HBA1 - GTEx") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](TPM)))

gtex_plot_HBA1


cor_hba1_gtex= e$Raw$GTEx %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "HBA1") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$GTEx, by= "SampleID") %>% 
  mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  ))) %>% 
  mutate(logExp = log10(Exp)) %>%
  ggscatter(., x = "PMI.hours", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("HBA1 - GTEx") +
  ylab("TPM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")

```

```{r}
bspan_plot_HBA1 = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "HBA1") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
   mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
  ggplot(aes(x = PMI.intervals,  y = log10(Exp), fill = Regions)) + geom_boxplot() +
  theme_bw() + 
  scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + 
  xlab ("PMI Intervals") + ggtitle("HBA1 - BrainSpan") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab(bquote(log["10"](RPKM)))
bspan_plot_HBA1


cor_hba1_bspan = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "HBA1") %>% 
    melt() %>% 
  dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan, by= "SampleID") %>% 
    mutate(PMI.intervals = factor(PMI.intervals, levels = c(
    "< 1",
   "\u2265 1 & < 4", 
   "\u2265 4 & < 8", 
    "\u2265 8 & < 12", 
    "\u2265 12 & < 16", 
    "\u2265 16 & < 20", 
     "\u2265 20"
  )))%>%
  drop_na(PMI.intervals) %>%
    ggscatter(., x = "PMI", y = "Exp", 
            add = "reg.line", 
            conf.int = TRUE, color = "Regions", palette = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) +
  stat_cor(aes(color = Regions), label.x = 3) + 
   scale_color_manual(values = c("Cortex" = "#eb103a", 
                                 "Subcortex" = "#333fa3", 
                                 "Cerebellum" = "#e3ae1e",
                                 "Spinal Cord" = "#778012", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Forebrain and midbrain" = "#9D696C"
                                )) + theme_bw() + xlab("PMI (in hours)") + ggtitle("HBA1 - BrainSpan") +
  ylab("RPKM") +  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")

```


```{r}
gtex_var_hba1 = var_part$GTEx %>% 
  dplyr::filter(gene_name %in% "HBA1") %>% 
  melt() %>%
  ggplot(aes(x= fct_reorder(variable, value), y = value)) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + xlab("") + ggtitle("HBA1 - GTEx") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ))
```



```{r}
bspan_var_hba1 =var_part$BSpan %>% 
  dplyr::filter(gene_name %in% "HBA1") %>% 
  melt()  %>% 
  ggplot(aes(x= fct_reorder(variable, value), y = value)) + geom_bar(stat = "identity", fill = "#F04C51") +
  coord_flip() + theme_bw() + ylab("Variance Explained (%)") + xlab("") + ggtitle("HBA1 - BrainSpan") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
        axis.text.y = element_text(size = 15, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) 

```


```{r}
pdf(file = here::here("output/Thesis_plots/Variance/Examples/RNA-degradation.pdf"), height=20, width=16)
ggarrange(bspan_fig_RNASE2, gtex_fig_RNASE2, 
          bspan_fig, gtex_plot, 
          bspan_plot_HBA1, gtex_plot_HBA1, 
          bspan_plot_EGR3, gtex_plot_egr3, ncol=2, nrow=4)
dev.off()
```

```{r}
pdf(file = here::here("output/Thesis_plots/Variance/Examples/RNA-degradation_varPart.pdf"), height=15, width=10)
ggarrange(bspan_var_cxlc2, gtex_var_cxcl2, 
          bspan_var_hba1, gtex_var_hba1, 
          bspan_var_egr3, gtex_var_egr3, ncol=2, nrow=3)
dev.off()

```

```{r}

#pdf(file = here::here("output/Thesis_plots/Variance/Examples/RNA-degradation_cor.pdf"), height=20, width=16)
ggarrange(cor_RNASE2_bspan, cor_RNASE2_gtex, 
          cor_CXCL2_bspan, cor_CXCL2_gtex,
          cor_hba1_bspan, cor_hba1_gtex, 
          cor_EGR3_bspan, cor_egr3_gtex, ncol =2, nrow=4)

ggsave(filename = here::here("output/Thesis_plots/Variance/Examples/RNA-degradation_cor.png"), height=20, width=16, dpi=1200)
```




## Understanding spatiotemporal properties of Neurodevelopmental disorder genes

```{r}
genes_of_interest = c("FMR1", "ZMYND8", "UPF1", "SPAST", "ATL1", "CTNNB1", "KCNQ2", "SPR", "UPF3B", "UPF3A")
exp_data = list()
datasets = c("BSpan", "GTEx", "BSeq", "PE", "HDBR")
for (x in datasets) {
  message(paste("Subsetting samples for", x))
  dataset = e$Raw[[x]]
  md_sub = md[[x]]
  name = x
  sub = dataset %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% genes_of_interest) %>% 
    melt() %>%
    dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    mutate(Dataset = x) %>% 
    left_join(md_sub %>% 
                dplyr::select(SampleID, Regions,
                              StructureAcronym, Period, 
                              AgeInterval), by = "SampleID")
  exp_data[[paste0(name)]] = sub 
  
}
```


### Cerebral Palsy genes


- SPAST

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPAST" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPAST" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```


```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPAST" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
   geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPAST" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```


```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPAST" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/SPAST-all.pdf"),
       height = 14, width = 16)
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```

- CTNNB1

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "CTNNB1" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "CTNNB1" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```


```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "CTNNB1" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
   geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "CTNNB1" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```


```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "CTNNB1" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/CTNNB1-all.pdf"),
       height = 14, width = 16)
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```

- ATL1

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ATL1" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ATL1" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```


```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ATL1" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
   geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ATL1" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```


```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ATL1" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/ATL1-all.pdf"),
       height = 14, width = 16)
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```

- SPR

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPR" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPR" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

```


```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPR" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
   geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))


```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPR" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  + 
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```


```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "SPR" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/SPR.pdf"),
       height = 14, width = 16)
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```



- Brain structures in cerebral palsy genes

```{r}
cp_genes_structures = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan"& Regions == "Cortex")) %>%
  dplyr::filter(gene_name %in% c( "CTNNB1")) %>%
  dplyr::filter(!StructureAcronym %in% c("TCx", "Ocx", "MGE",
                                         "M1C-S1C", "PCx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = StructureAcronym)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw()  + ggtitle("BrainSpan") +
  #scale_fill_manual(values = c("AMY" = "#d8e2dc", 
  #                             "HIP" = "#ffe5d9", 
   #                            "MD" = "#f4acb7",
   #                            "STR" = "#9d8189")) +
  xlab("") + facet_wrap(~StructureAcronym, scales = "free") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM))) 


pdf(file = here::here("output/Thesis_plots/Examples/NDDs/CP_genes_structure.pdf"),
       height = 16, width = 16)
cp_genes_structures 
dev.off()
```


```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
 
bspan

```

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bseq

```

```{r}
fmr1_bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FMR1" & Dataset == "BSpan"& Regions == "Subcortex")) %>%
  dplyr::filter(!StructureAcronym %in% c("CGE", "DTH", "LGE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = StructureAcronym)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw()  + ggtitle("BrainSpan") +
  scale_fill_manual(values = c("AMY" = "#d8e2dc", 
                               "HIP" = "#ffe5d9", 
                               "MD" = "#f4acb7",
                               "STR" = "#9d8189")) +
  xlab("") + facet_wrap(~StructureAcronym) +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM))) 

```



```{r}
ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/FMR1-BSpan-structures.svg"),
       height = 10, width = 16, units = "in", plot = fmr1_bspan)


```




- ZMYND8 

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr
```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

pe

```


```{r}
ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/ZMYND8-all.svg"),
       height = 14, width = 16, units = "in")
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```


## FMR1
```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.05) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)
```

```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)
```





## FMR1
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bspan
```


```{r}
fmr1_cor =exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan"& Regions == "Cortex")) %>%
  dplyr::filter(gene_name %in% c( "FMR1")) %>%
  dplyr::filter(!StructureAcronym %in% c("TCx", "Ocx", "MGE",
                                         "M1C-S1C", "PCx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = StructureAcronym)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw()  + ggtitle("BrainSpan") +
 # scale_fill_manual(values = c("AMY" = "#d8e2dc", 
  #                             "HIP" = "#ffe5d9", 
   #                           "MD" = "#f4acb7",
    #                           "STR" = "#9d8189")) +
  xlab("") + facet_wrap(~StructureAcronym, scales = "free") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM))) + scale_fill_viridis(discrete = TRUE)

ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/FMR1-BSpan-cor-structures.svg"),
       height = 10, width = 16, units = "in", plot = fmr1_cor)

```


```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
    theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
bseq

```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
gtex

```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr

```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
pe

```

```{r}

ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/FMR1-all.png"),
       height = 14, width = 16, units = "in", dpi=1200)
```

## UPF1

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bspan
```


```{r}
upf1_cor =exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan"& Regions == "Cortex")) %>%
  dplyr::filter(gene_name %in% c( "UPF1")) %>%
  dplyr::filter(!StructureAcronym %in% c("TCx", "Ocx", "MGE",
                                         "M1C-S1C", "PCx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = StructureAcronym)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
   theme_bw()  + ggtitle("BrainSpan") +
 # scale_fill_manual(values = c("AMY" = "#d8e2dc", 
  #                             "HIP" = "#ffe5d9", 
   #                           "MD" = "#f4acb7",
    #                           "STR" = "#9d8189")) +
  xlab("") + facet_wrap(~StructureAcronym, scales = "free") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM))) + scale_fill_viridis(discrete = TRUE)

ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/FMR1-BSpan-cor-structures.svg"),
       height = 10, width = 16, units = "in", plot = fmr1_cor)

```


```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
    theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
bseq

```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
gtex

```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr

```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
pe

```


```{r}
ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/UPF1-all.png"),
       height = 14, width = 16, units = "in")
ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

dev.off()
```

# UPF3B and UPF3B

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "UPF3B")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bspan
```




```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "UPF3B")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
    theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
bseq

```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "UPF3B")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
gtex

```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "UPF3B")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr

```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "UPF3B")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
pe

```


```{r}

ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/UPF3B-all.png"),
       height = 14, width = 16, units = "in", dpi =1200)
```


```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "UPF3A")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
   theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSpan") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))

bspan
```




```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "UPF3A")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
    theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("BrainSeq") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 3.5,lty = "dashed") +
  ylab(bquote(log["10"](RPKM)))
bseq

```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "UPF3A")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("GTEx") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 0.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
gtex

```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "UPF3A")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("HDBR") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 6.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))

hdbr

```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "UPF3A")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  + ggtitle("PsychENCODE") +
  xlab("") +
  #facet_grid(Dataset~gene_name, scales = "free") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top")  +
  ylab("log10(RPKM)") + geom_vline(xintercept = 5.5,lty = "dashed") +
  ylab(bquote(log["10"](TPM)))
pe

```


```{r}

ggarrange(bseq, bspan,
          gtex, hdbr, pe, ncol=2, nrow=3 )

ggsave(filename = here::here("output/Thesis_plots/Examples/NDDs/UPF3A-all.png"),
       height = 14, width = 16, units = "in", dpi =1200)
```


## Cell-type deconvolution 

```{r}
decon_res = list()
decon_res$BSpan = read.csv("/home/neuro/Documents/BrainData/Bulk/BrainSpan/Formatted/BrainSpan-decon.csv", header=TRUE)

decon_res$GTEx = read.csv("/home/neuro/Documents/BrainData/Bulk/GTEx/Formatted/GTEx-decon.csv", header=TRUE)
```

```{r}
bspan_prop_zmynd8 =decon_res$BSpan %>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$BSpan %>% 
              dplyr::filter(EnsemblID == "ENSG00000101040"), by = "SampleID") %>%
    ggplot(aes(x=Neurons, y = log10(Exp), color = Regions, shape = Period)) + geom_point(alpha = 0.75, size =6) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + ylab(bquote(log["10"](RPKM))) +
  ggtitle("ZMYND8 - BrainSpan")


```


```{r}
GTEx_prop_zmynd8 =decon_res$GTEx%>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$GTEx %>% 
              dplyr::filter(EnsemblID == "ENSG00000101040"), by = "SampleID") %>%
  ggplot(aes(x=Neurons, y = log10(Exp), color = Regions)) + geom_point(alpha = 0.75, size =6) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + ylab(bquote(log["10"](TPM))) +
  ggtitle("ZMYND8 - GTEx") 

GTEx_prop_zmynd8 

```

```{r}
bspan_prop_FMRI =decon_res$BSpan %>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$BSpan %>% 
              dplyr::filter(EnsemblID == "ENSG00000102081"), by = "SampleID") %>%
    ggplot(aes(x=Neurons, y = log10(Exp), color = Regions, shape = Period)) + geom_point(alpha = 0.75, size =6) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + ylab(bquote(log["10"](RPKM))) +
  ggtitle("FMR1 - Brainspan")
```


```{r}
GTEx_prop =decon_res$GTEx%>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$GTEx %>% 
              dplyr::filter(EnsemblID == "ENSG00000102081"), by = "SampleID") %>%
  ggplot(aes(x=Neurons, y = log10(Exp), color = Regions)) + geom_point(alpha = 0.75, size =5) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
     theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + ylab(bquote(log["10"](TPM))) +
  ggtitle("FMR1 - GTEx")
GTEx_prop
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/NDD-prop-all.pdf"),
       height = 10, width = 12)

ggarrange(bspan_prop_zmynd8,GTEx_prop_zmynd8,
             bspan_prop_FMRI,GTEx_prop, ncol =2, nrow=2)



dev.off()
```










# Single-nucleus and deconvolution exploration of bulk RNA-seq expression


```{r}
sn.dir =file.path("/home/neuro/Documents/BrainData/single-cell")

snRNA = list()

snRNA$hca = read.csv(file.path(sn.dir, "hca/Processed/HCA-exp.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
snRNA$vel = read.csv(file.path(sn.dir, "velmeshev/FormattedData/Velmeshev-exp.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)

snRNA$cer = read.csv(file.path(sn.dir, "aldinger/Aldringer-exp.csv"), header=TRUE, 
                     check.names = FALSE, row.names = 1)
#snRNA$cameron 

sn_md = list()

sn_md$hca = read.csv(file.path(sn.dir, "hca/Processed/HCA-metadata.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
sn_md$vel = read.csv(file.path(sn.dir, "velmeshev/FormattedData/Velmeshev-metadata.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
sn_md$cer = read.csv(file.path(sn.dir, "aldinger/Aldringer-metadata.csv"), 
                     header=TRUE, check.names =FALSE)

sn_md$cer = read.csv(file.path(sn.dir, "cameron/Cameron-metadata.csv"), 
                     header=TRUE, check.names =FALSE, row.names = 1)


colnames(sn_md$hca)[9:13] = c("MajorCellType", "Age", 
                              "AgeNumeric", "AgeInterval", "Regions")

colnames(sn_md$vel)[c(6,8)] = c("Age", "Diagnosis")
colnames(sn_md$vel)[c(19,20)] = c("AgeInterval", "Regions") 
write.csv(sn_md$hca, "/home/neuro/Documents/BrainData/single-cell/hca/Processed/HCA-metadata.csv")

write.csv(sn_md$vel, "/home/neuro/Documents/BrainData/single-cell/velmeshev/FormattedData/Velmeshev-metadata.csv")

```

```{r}
snRNA$vel[,-1] %>% 
  rownames_to_column("genes") %>% 
  dplyr::filter(genes == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$vel %>% 
              rownames_to_column("CellID"), by ="CellID") %>% 
  ggplot(aes(x=MajorCelltype, y = log10(value), fill = diagnosis)) + geom_boxplot() + 
  theme_bw()
```

```{r}
snRNA$vel[,-1] %>% 
  rownames_to_column("genes") %>% 
  dplyr::filter(genes == "ZMYND8") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$vel %>% 
              rownames_to_column("CellID"), by ="CellID") %>% 
  ggplot(aes(x=MajorCelltype, y = log10(value), fill = diagnosis)) + geom_boxplot() + 
  theme_bw()
```

```{r}
vel_fmr1 = snRNA$vel %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$vel, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = Diagnosus)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("FMR1") +
  scale_fill_manual(values=c("Control"= "#61bdbd",
                             "ASD" = "#ee6d6d")) + ggtitle("FMR1 - Velmeshev") + labs(fill = "Diagnsosis")
```

```{r}
hca_fmr1 = snRNA$hca %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("FMR1 - HCA") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99")) + labs(fill  ="Major Cell Type")

```


```{r}
cam_fmr1 = snRNA$cam[,-1] %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$cer %>%
              dplyr::rename("CellID" = "cellID"), by ="CellID") %>% 
  ggplot(aes(x= CellTypeGroup, y = log10(value), fill =  CellTypeGroup)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("FMR1 - Cameron dataset") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPC" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Endothelial" = "#ffcf99", 
                               "Cajal-Retzius cell" = "#6d6875",
                               "Cycling progenitor cell" = "#4a4e69",
                               "Radial glia" = "#e9edc9",
                               "Intermediate progenitor" = "#fefae0",
                               "Undefined class Neurons" = "#e5e5e5")) + 
  labs(fill  ="Major Cell Type") + ylim(0, 3.5)

```

```{r}
pdf(file = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Thesis_plots/Examples/NDDs/CT-exp-FMR1.pdf", height = 8, width =18)
cam_fmr1 + hca_fmr1
dev.off()
```


## ZMYND8 


```{r}
hca_zmynd8 = snRNA$hca %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("ZMYND8 - HCA") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99")) + labs(fill  ="Major Cell Type")

```


```{r}
cam_zmynd8 = snRNA$cam[,-1] %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$cer %>%
              dplyr::rename("CellID" = "cellID"), by ="CellID") %>% 
  ggplot(aes(x= CellTypeGroup, y = log10(value), fill =  CellTypeGroup)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("ZMYND8 - Cameron dataset") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPC" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Endothelial" = "#ffcf99", 
                               "Cajal-Retzius cell" = "#6d6875",
                               "Cycling progenitor cell" = "#4a4e69",
                               "Radial glia" = "#e9edc9",
                               "Intermediate progenitor" = "#fefae0",
                               "Undefined class Neurons" = "#e5e5e5")) + 
  labs(fill  ="Major Cell Type") + ylim(0, 3.5)

```

```{r}
pdf(file = "/home/neuro/Documents/Brain_integrative_transcriptome/BITHub-preprocessing/output/Thesis_plots/Examples/NDDs/CT-exp-ZMYND8.pdf", height = 8, width =18)
cam_zmynd8 + hca_zmynd8
dev.off()
```



## Cerebral palsy genes 

- HCA (postnatal)
```{r}
ctnnb1 = snRNA$hca %>% 
  dplyr::filter(Symbol == "CTNNB1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("CTNNB1") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99"))
```


```{r}
spast =snRNA$hca %>% 
  dplyr::filter(Symbol == "SPAST") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("SPAST") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99"))
```


```{r}
spr = snRNA$hca %>% 
  dplyr::filter(Symbol == "SPR") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("SPR") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99"))
```

```{r}
atl1 =snRNA$hca %>% 
  dplyr::filter(Symbol == "ATL1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
    theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("ATL1") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99"))
```

```{r}
pdf(file = here::here("output/Thesis_plots/Examples/NDDs/CP-cell-types-HCA.pdf"),
       height = 14, width = 16)
ggarrange(atl1, ctnnb1, spast, spr, 
          ncol = 2, nrow = 2, common.legend = TRUE)
dev.off()
```



## FMR1


```{r}
snRNA$hca %>% 
  dplyr::filter(Symbol == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
    theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("FMR1") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99"))
```


```{r} 
snRNA$hca %>% 
  dplyr::filter(Symbol == "GFAP") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$hca, by ="CellID") %>% 
  ggplot(aes(x=MajorCellType, y = log10(value), fill = MajorCellType)) + geom_boxplot() + 
  theme_bw() +  theme_bw() + theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top") + xlab("") +  ylab(bquote(log["10"](CPM))) + ggtitle("GFAP - HCA") + 
  scale_fill_manual(values = c("Astrocytes" = "#778da9",
                                "Microglia" = "#c89f9c", 
                                "Oligodendrocytes" = "#d5c6e0",
                                "OPCs" = "#5e548e",
                                "Excitatory Neurons" = "#ffa69e", 
                                "Inhibitory Neurons" = "#e26d5c",
                               "Vasculature" = "#ffcf99")) + labs(fill  ="Major Cell Type")
```



### DIPC2 paper


```{r}
md$BSpan$Regions_additional[md$BSpan$StructureAcronym == "OFC" | 
                           md$BSpan$StructureAcronym  == "DLPFC" |
                           md$BSpan$StructureAcronym == "VFC" |
                             md$BSpan$StructureAcronym == "ACC" |
                           md$BSpan$StructureAcronym  ==  "MFC" | 
                           md$BSpan$StructureAcronym  == "M1C" |  ## frontal cortex
                           md$BSpan$StructureAcronym  == "M1C-S1C" | 
                           md$BSpan$StructureAcronym  == "IPC" | 
                           md$BSpan$StructureAcronym  == "S1C" |  ### parietal cortex 
                          md$BSpan$StructureAcronym  == "A1C" | 
                           md$BSpan$StructureAcronym  == "STC" |  
                           md$BSpan$StructureAcronym  == "ITC" | ##temporal cortex
                           md$BSpan$StructureAcronym  == "V1C" |
                           md$BSpan$StructureAcronym == "Ocx" |
                           md$BSpan$StructureAcronym  == "PCx" | 
                           md$BSpan$StructureAcronym  == "TCx" |
                            md$BSpan$StructureAcronym  == "MGE"] <- "NCX"


md$BSpan$Regions_additional[
                           md$BSpan$StructureAcronym  == "LGE" | 
                           md$BSpan$StructureAcronym  == "CGE" | 
                           md$BSpan$StructureAcronym  == "STR" |
                           md$BSpan$StructureAcronym  ==  "HIP" | 
                           md$BSpan$StructureAcronym  == "AMY" |
                           md$BSpan$StructureAcronym  == "DTH" | 
                           md$BSpan$StructureAcronym  == "MD" |
                           md$BSpan$StructureAcronym  == "URL" | 
                          md$BSpan$StructureAcronym  == "CBC" | 
                           md$BSpan$StructureAcronym == "CB"] <- "Other_regions"


```

```{r}


sub = e$Raw$BSpan %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% "DIP2C") %>% 
    melt() %>%
    dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    left_join(md$BSpan %>% 
                dplyr::select(SampleID, Regions,
                              StructureAcronym, Period, 
                              AgeInterval, Regions_additional), by = "SampleID")

```


```{r}
ncx = sub %>%
  mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>%
  dplyr::filter(Regions_additional == "NCX") %>%
  ggplot(aes(AgeInterval, Exp))  + 
  geom_boxplot(aes(fill=AgeInterval), alpha=0.65)  + 
  geom_jitter(size=2.5) +
  scale_color_viridis(discrete = TRUE, option = "magma")+ theme_bw() +
  scale_fill_viridis(discrete = TRUE, direction = 1)  +
  xlab("Stage") + ylab("RPKM") +
  theme(legend.position = "none", 
        axis.text=element_text(size=11, face="bold", color="black"), 
        axis.title = element_text(siz=12, face="bold", color="black"), 
        axis.text.x = element_text(angle=90)) +
  ggtitle("Neocortex") + 
    theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "none") + 
  geom_vline(xintercept = 6.5,lty = "dashed") 


other_regions = sub %>%
  mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>%
  dplyr::filter(Regions_additional == "Other_regions") %>%
  ggplot(aes(AgeInterval, Exp))  + 
  geom_boxplot(aes(fill=AgeInterval), alpha=0.65)  + 
  geom_jitter(size=2.5) +
  scale_color_viridis(discrete = TRUE, option = "magma")+ theme_bw() +
  scale_fill_viridis(discrete = TRUE, direction = 1)  +
  xlab("Stage") + ylab("RPKM") +
  theme(legend.position = "none", 
        axis.text=element_text(size=11, face="bold", color="black"), 
        axis.title = element_text(siz=12, face="bold", color="black"), 
        axis.text.x = element_text(angle=90)) +
  ggtitle("Other regions") + 
    theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "none") + 
  geom_vline(xintercept = 6.5,lty = "dashed") 
```


```{r}
prenatal =sub %>%  
  dplyr::filter(Period == "Prenatal") %>%
  ggplot(aes(StructureAcronym, Exp))  + 
  geom_boxplot(aes(fill=StructureAcronym), alpha=0.65)  + 
  geom_jitter(size=2.5) +
  scale_color_viridis(discrete = TRUE, option = "magma")+ theme_bw() +
  scale_fill_viridis(discrete = TRUE, direction = 1)  +
  xlab("Region") + ylab("RPKM") + ylim(0, 30) +
  theme(legend.position = "none", 
        axis.text=element_text(size=11, face="bold", color="black"), 
        axis.title = element_text(siz=12, face="bold", color="black"), 
        axis.text.x = element_text(angle=90)) + ggtitle("Prenatal Expression") + 
      theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "none") 
```

```{r}
postnatal =sub %>%  
  dplyr::filter(Period == "Postnatal") %>%
  ggplot(aes(StructureAcronym, Exp))  + 
  geom_boxplot(aes(fill=StructureAcronym), alpha=0.65)  + 
  geom_jitter(size=2.5) +
  scale_color_viridis(discrete = TRUE, option = "magma")+ theme_bw() +
  scale_fill_viridis(discrete = TRUE, direction = 1)  +
  xlab("Region") + ylab("RPKM") + ylim(0, 30) +
  theme(legend.position = "none", 
        axis.text=element_text(size=11, face="bold", color="black"), 
        axis.title = element_text(siz=12, face="bold", color="black"), 
        axis.text.x = element_text(angle=90)) + ggtitle("Prenatal Expression") + 
      theme(axis.text.x = element_text(size = 12, family = "serif", color = "black", angle = 90),
        axis.text.y = element_text(size = 12, family = "serif", color = "black"), 
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "none") 
```


```{r}
pdf(file = here::here("output/Thesis_plots/Examples/DIP2C-regions.pdf"),
       height = 6, width = 15)
ggarrange(ncx, other_regions, ncol=2)

dev.off()


pdf(file = here::here("output/Thesis_plots/Examples/DIP2C-development.pdf"),
       height = 6, width = 15)
ggarrange(prenatal, postnatal, ncol=2)

dev.off()
```


```{r}
prenatal_structures <- sub %>%  
  dplyr::filter(Period == "Prenatal")


model_pren=lm(prenatal_structures$Exp ~ prenatal_structures$StructureAcronym); aov(model_pren)
summary(aov(model_pren))


##### postnatal

postnatal_structures <- ZMYND8_data %>%  
  dplyr::filter(period == "postnatal") %>%
  dplyr::filter(structure_acronym %in% as.character(postn_R_structures$Var1))

model_postn=lm(postnatal_structures$value ~ postnatal_structures$structure_acronym)
summary(aov(model_postn))
```
