---
title: "gene-exp"
author: "urwahnawaz"
date: "2023-10-03"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

```{r libraries include=FALSE}
libs = c("dplyr", "ggplot2", "reshape2", "tools", "magrittr", "tibble", "readxl", 
         "data.table", "scales", "tidyr", "reshape2", "stringr", "tidyverse", "readxl", "corrplot", "viridis", 
         "sunburstR", "pheatmap")
libsLoaded <- lapply(libs,function(l){suppressWarnings(suppressMessages(library(l, character.only = TRUE)))})
library(EnsDb.Hsapiens.v86)
source(here::here("code/preprocess/functions.R"))
source(here::here("code/preprocess/def_stages.R"))
```

```{r annotations}
gene2symbol = genes(EnsDb.Hsapiens.v86, return.type="DataFrame") %>%
    as.data.frame()
gene2symbol %<>% 
    dplyr::select("EnsemblID" = "gene_id", gene_name)

```

```{r load-exp}
brain.dir =file.path("/home/neuro/Documents/BrainData/Bulk")
e <- list()
  
e$GTEx <- read.csv(file.path(brain.dir, "/GTEx/Formatted/GTEx-exp.csv"), 
                       check.names = FALSE, row.names = 1) 
e$PE <- read.csv(file.path(brain.dir, "/PsychEncode/Formatted/PsychEncode-exp.csv"), row.names = 1,
                 check.names = FALSE) 
e$BSpan <- read.csv(file.path(brain.dir, "/BrainSpan/Formatted/BrainSpan-exp.csv"),
                        check.names = FALSE, row.names=1)
e$BSeq <- read.csv(file.path(brain.dir,"/Brainseq/Formatted/BrainSeq-exp.csv"), 
                       check.names = FALSE, row.names = 1)

e$HDBR <- read.csv(file.path(brain.dir, "/HDBR/Formatted/HDBR-exp.csv"), 
                       check.names = FALSE, row.names = 1) %>% 
  column_to_rownames("EnsemblID")


```

```{r load-md}
md = list()
md$GTEx <- read.csv(file.path(brain.dir, "/GTEx/Formatted/GTEx-metadata.csv"), 
                       check.names = FALSE, row.names = 1)
md$PE <- read.csv(file.path(brain.dir,"/PsychEncode/Formatted/PsychEncode-metadata.csv"),
                  check.names = FALSE)
md$BSeq <- read.csv(file.path(brain.dir,"/Brainseq/Formatted/BrainSeq-metadata.csv"), 
                       check.names = FALSE)
md$HDBR <- read.csv(file.path(brain.dir, "/HDBR/Formatted/HDBR-metadata.csv"), 
                       check.names = FALSE, row.names = 1) 
md$Ram <- read.csv(file.path(brain.dir, "/Ramakar/Formatted/Ramaker-metadata.csv"),
                       check.names = FALSE, row.names = 1)

md$BSpan <- read.csv(file.path(brain.dir,
                               "/BrainSpan/Formatted/BrainSpan-metadata.csv"),
                     check.names = FALSE)
```


```{r}
genes_of_interest = c("FMR1", "ZMYND8", "UPF1", "FOXG1", "ZIC2", "MKI67", "KIF1B")
exp_data = list()
datasets = c("BSpan", "GTEx", "BSeq", "PE", "HDBR")
for (x in datasets) {
  message(paste("Subsetting samples for", x))
  dataset = e[[x]]
  md_sub = md[[x]]
  name = x
  sub = dataset %>% 
    as.data.frame() %>%
    rownames_to_column("EnsemblID") %>%
    left_join(gene2symbol, by = "EnsemblID") %>% 
    dplyr::filter(gene_name %in% genes_of_interest) %>% 
    melt() %>%
    dplyr::rename("SampleID" = "variable", "Exp" = "value") %>% 
    mutate(Dataset = x) %>% 
    left_join(md_sub %>% 
                dplyr::select(SampleID, Regions,
                              StructureAcronym, Period, 
                              AgeInterval), by = "SampleID")
  exp_data[[paste0(name)]] = sub 
  
}
```

## Plots for ZIC2 


- BrainSeq 

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 2) +
  ylab("log10(RPKM)")

bseq
ggsave(bseq, file=here::here("output/Thesis_plots/Examples/BrainSeq-ZIC2.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```

- BrainSpan
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 2.5) +
  ylab("log10(RPKM)")
bspan
ggsave(bspan, file=here::here("output/Thesis_plots/Examples/BrainSpan-ZIC2.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- GTEx

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 3) +
  ylab("log10(RPKM)")
gtex
ggsave(gtex, file=here::here("output/Thesis_plots/Examples/GTEx-ZIC2.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- HDBR 

```{r}
hdbr  = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 2.5) +
  ylab("log10(RPKM)")

hdbr
ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/HDBR-ZIC2.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- PsychEncode 

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "ZIC2" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 2) +
  ylab("log10(RPKM)")

pe
ggsave(pe, file=here::here("output/Thesis_plots/Examples/PsychEncode-ZIC2.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```

## MKI67

- BrainSeq 

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 2) +
  ylab("log10(RPKM)")
bseq
ggsave(bseq, file=here::here("output/Thesis_plots/Examples/BrainSeq-MKI67.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```

- BrainSpan
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 2) +
  ylab("log10(RPKM)")

bspan
ggsave(bspan, file=here::here("output/Thesis_plots/Examples/BrainSpan-MKI67.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- GTEx

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 2) +
  ylab("log10(RPKM)")

gtex
ggsave(gtex, file=here::here("output/Thesis_plots/Examples/GTEx-MKI67.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- HDBR 

```{r}
hdbr  = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-2, 2.5) +
  ylab("log10(RPKM)")

hdbr
ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/HDBR-MKI67.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- PsychEncode 

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "MKI67" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 2) +
  ylab("log10(RPKM)")

pe
ggsave(pe, file=here::here("output/Thesis_plots/Examples/PsychEncode-MKI67.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```




## FOXG1 
- BrainSeq 

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FOXG1" & Dataset == "BSeq")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 6) +
  ylab("log10(RPKM)")

ggsave(bseq, file=here::here("output/Thesis_plots/Examples/BrainSeq-FOXG1.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```

- BrainSpan
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FOXG1" & Dataset == "BSpan")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 6) +
  ylab("log10(RPKM)")

ggsave(bspan, file=here::here("output/Thesis_plots/Examples/BrainSpan-FOXG1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- GTEx

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FOXG1" & Dataset == "GTEx")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 6) +
  ylab("log10(RPKM)")

ggsave(gtex, file=here::here("output/Thesis_plots/Examples/GTEx-FOXG1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- HDBR 

```{r}
hdbr  = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FOXG1" & Dataset == "HDBR")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 6) +
  ylab("log10(RPKM)")


ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/HDBR-FOXG1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

- PsychEncode 

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FOXG1" & Dataset == "PE")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log10(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), linewidth =0.3) + 
  theme_bw() + scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                ))  +
  #facet_grid(Dataset~gene_name, scales = "free") + 
  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(-3, 6) +
  ylab("log10(RPKM)")


ggsave(pe, file=here::here("output/Thesis_plots/Examples/PsychEncode-FOXG1.svg"), 
        width = 8, 
       height =5, 
       units = "in")

```



```{r}

sub = snRNA$hca %>% 
    as.data.frame() %>%
    dplyr::filter(Symbol %in% "MKI67") %>% 
    melt() %>%
    dplyr::rename("CellID" = "variable", "Exp" = "value")

#%>% 
sub %<>% 
    left_join(sn_md$hca %>%
                rownames_to_column("CellID"),  by = "CellID")


ggplot(sub, aes(x = MajorCelltype, y = log10(Exp), 
                fill = MajorCelltype))  +geom_boxplot() + theme_bw()

```




## ZMYND8 gene 

```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain"))+ theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 3) +
  ylab("log10(RPKM)")

bspan
ggsave(bspan, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSpan-ZMYND8.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 3.6) +
  ylab("log10(RPKM)")

bseq
ggsave(bseq, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSeq-ZMYND8.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain"))  + theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 4.5) +
  ylab("log10(RPKM)")
gtex
ggsave(gtex, file=here::here("output/Thesis_plots/Examples/NDDs/GTEx-ZMYND8.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 4) +
  ylab("log10(RPKM)")  
hdbr
ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/NDDs/HDBR-ZMYND8.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "ZMYND8")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 4.5) +
  ylab("log10(RPKM)")

pe
ggsave(pe, file=here::here("output/Thesis_plots/Examples/NDDs/PsychEncode-ZMYND8.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```



```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.05) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)
```

```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)
```


## FMR1
```{r}
bspan = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSpan" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain"))+ theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 3.5) +
  ylab("log10(RPKM)")

bspan
ggsave(bspan, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSpan-FMR1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
bseq = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "BSeq" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +  theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 3.5) +
  ylab("log10(RPKM)")
bseq
ggsave(bseq, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSeq-FMR1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
gtex = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "GTEx" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain"))  + theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 4.5) +
  ylab("log10(RPKM)")

gtex
ggsave(gtex, file=here::here("output/Thesis_plots/Examples/NDDs/GTEx-FMR1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
hdbr = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "HDBR" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 4.5) +
  ylab("log10(RPKM)")  

hdbr
ggsave(hdbr, file=here::here("output/Thesis_plots/Examples/NDDs/HDBR-FMR1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
pe = exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((Dataset == "PE" & gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +w
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
    theme(
        axis.text.x = element_text(angle = 90), 
        legend.position = "top", 
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 )) + ylim(0, 5) +
  ylab("log10(RPKM)")

pe
ggsave(pe, file=here::here("output/Thesis_plots/Examples/NDDs/PsychEncode-FMR1.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```



```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "FMR1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.05) + 
  theme_bw() +scale_fill_manual(values= c("Cerebellum" = "#FFD256",
                                           "Subcortex" = "#838DE0",                 
                                          "Cortex" = "#E85571", 
                                           "Spinal Cord" = "#C6D14A", 
                                           "Forebrain" = "#F6B19D",
                                           "Midbrain" = "#AC93AD", 
                                           "Hindbrain" = "#6E8B8E",
                                          "Brain fragment" = "#FFA6B7", 
                                          "Forebrain and midbrain")) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)
```

```{r}
exp_data %>% 
  do.call(rbind, .) %>% 
  dplyr::filter((gene_name == "UPF1")) %>%
    mutate(AgeInterval = factor(AgeInterval, 
                              levels = c("4-7pcw", "8-9pcw",
                                         "10-12pcw", "13-15pcw", "16-18pcw",
                                         "19-24pcw", "25-38pcw", "0-5mos",
                                         "6-18mos", "19mos-5yrs", "6-11yrs",
                                         "12-19yrs", "20-29yrs", "30-39yrs",
                                         "40-49yrs", "50-59yrs", "60-69yrs",
                                         "70-79yrs", "80-89yrs", "90-99yrs"))) %>% 
  drop_na() %>%
  ggplot(aes(x = AgeInterval, y =log(Exp), fill = Regions)) +
# geom_violin(position = position_dodge(0.8), color = NA) +
  geom_boxplot(position = position_dodge(0.8), outlier.shape = NA, linewidth = 0.3) + 
  theme_bw() +scale_fill_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) +
  facet_grid(Dataset~gene_name, scales = "free") + 
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90)) + ylim(-3, 6)


```

## Cell-type deconvolution 

```{r}
decon_res = list()
decon_res$BSpan = read.csv("/home/neuro/Documents/BrainData/Bulk/BrainSpan/Formatted/BrainSpan-decon.csv", header=TRUE)

decon_res$GTEx = read.csv("/home/neuro/Documents/BrainData/Bulk/GTEx/Formatted/GTEx-decon.csv", header=TRUE)
```

```{r}
bspan_prop =decon_res$BSpan %>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$BSpan %>% 
              dplyr::filter(EnsemblID == "ENSG00000101040"), by = "SampleID") %>%
    ggplot(aes(x=Neurons, y = log10(Exp), color = Regions, shape = Period)) + geom_point(alpha = 0.75, size =3) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
  theme(axis.text = element_text(size =12, face="bold"))


bspan_prop 
ggsave(bspan_prop, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSpan-ZMYND8-neuronal.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```


```{r}
GTEx_prop =decon_res$GTEx%>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$GTEx %>% 
              dplyr::filter(EnsemblID == "ENSG00000101040"), by = "SampleID") %>%
  ggplot(aes(x=Neurons, y = log10(Exp), color = Regions)) + geom_point(alpha = 0.75, size =3) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
  theme(axis.text = element_text(size =12, face="bold"))
GTEx_prop 
ggsave(GTEx_prop, file=here::here("output/Thesis_plots/Examples/NDDs/GTEx-ZMYND8-neuronal.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```

```{r}
bspan_prop =decon_res$BSpan %>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$BSpan %>% 
              dplyr::filter(EnsemblID == "ENSG00000102081"), by = "SampleID") %>%
    ggplot(aes(x=Neurons, y = log10(Exp), color = Regions, shape = Period)) + geom_point(alpha = 0.75, size =3) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
  theme(axis.text = element_text(size =12, face="bold"))

ggsave(bspan_prop, file=here::here("output/Thesis_plots/Examples/NDDs/BrainSpan-FMR1-neuronal.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```


```{r}
GTEx_prop =decon_res$GTEx%>% 
  dplyr::rename("SampleID" = "X") %>% 
  left_join(exp_data$GTEx %>% 
              dplyr::filter(EnsemblID == "ENSG00000102081"), by = "SampleID") %>%
  ggplot(aes(x=Neurons, y = log10(Exp), color = Regions)) + geom_point(alpha = 0.75, size =3) + theme_bw() +
  scale_color_manual(values = c("Cortex" = "#E85571", 
                                 "Subcortex" = "#838DE0", 
                                 "Cerebellum" = "#FFD256",
                                 "Spinal Cord" = "#C6D14A", 
                                 "Forebrain" = "#F6B19D", #
                                 "Midbrain" = "#AC93AD",
                                 "Hindbrain" = "#6E8B8E",
                                 "Brain" = "#FFA6B7", 
                                 "Chroid plexus" = "#9D696C"
                                )) + xlab("Neuronal proportion") +
  theme(axis.text = element_text(size =12, face="bold"))

ggsave(GTEx_prop, file=here::here("output/Thesis_plots/Examples/NDDs/GTEx-FMR1-neuronal.svg"), 
        width = 8, 
       height =5, 
       units = "in")
```


# Single-nucleus and deconvolution exploration of bulk RNA-seq expression


```{r}
sn.dir =file.path("/home/neuro/Documents/BrainData/single-cell")

snRNA = list()

snRNA$hca = read.csv(file.path(sn.dir, "hca/Processed/HCA-exp.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
snRNA$vel = read.csv(file.path(sn.dir, "velmeshev/FormattedData/Velmeshev-exp.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
#snRNA$cameron 

sn_md = list()

sn_md$hca = read.csv(file.path(sn.dir, "hca/Processed/HCA-metadata.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)
sn_md$vel = read.csv(file.path(sn.dir, "velmeshev/FormattedData/Velmeshev-metadata.csv"), 
                     header=TRUE, check.names = FALSE, row.names = 1)

```

```{r}
snRNA$vel[,-1] %>% 
  rownames_to_column("genes") %>% 
  dplyr::filter(genes == "FMR1") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$vel %>% 
              rownames_to_column("CellID"), by ="CellID") %>% 
  ggplot(aes(x=MajorCelltype, y = log10(value), fill = diagnosis)) + geom_boxplot() + 
  theme_bw()
```

```{r}
snRNA$vel[,-1] %>% 
  rownames_to_column("genes") %>% 
  dplyr::filter(genes == "ZMYND8") %>% 
  melt() %>% 
  dplyr::rename("CellID"= "variable") %>% 
  left_join(sn_md$vel %>% 
              rownames_to_column("CellID"), by ="CellID") %>% 
  ggplot(aes(x=MajorCelltype, y = log10(value), fill = diagnosis)) + geom_boxplot() + 
  theme_bw()
```

